---
title: "Schroeter_phytophthora"
author: "Barry"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
theme: cosmo
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}

library(tidyverse)
library(WGCNA)
library(DESeq2)
library(data.table)
library(reshape2)
library(gplots)
options(stringsAsFactors = FALSE);

```


```{r,data1,fig.height=8,fig.width=8}

# http://ftp.ebi.ac.uk/ensemblgenomes/pub/release-52/plants/gtf/lupinus_angustifolius/Lupinus_angustifolius.LupAngTanjil_v1.0.52.gtf.gz

download.file("http://ziemann-lab.net/public/barry/3col.tsv.gz",destfile = "3col.tsv.gz")
c3 <- read.table("3col.tsv.gz")
head(c3)
x <- t(acast(c3, V1~V2, value.var="V3"))
head(x)

# sample sheet
sample_metadata = read.csv(file = "sample_info.csv")
head(sample_metadata)
colnames(x) <- sample_metadata[,2]
head(x)

# get rid of controls
x <- x[,16:ncol(x)]
head(x)
dim(x)
x <- x[which(rowMeans(x)>=10),]
dim(x)

rpm <- x / colSums(x) * 1000000

dim(rpm)

# downsample matrix genes randomly - only for testing
#samplesize <- round(nrow(rpm)/5)
#rpm <- rpm[sample(x=1:nrow(rpm),size = samplesize,replace = FALSE),]
#dim(rpm)

```

Correlation heatmaps

```{r,cor1}

mycor <- cor(rpm)

heatmap.2(mycor,trace="none",scale="none")

genecor <- cor(t(rpm))

#heatmap.2(genecor,trace="none",scale="none")


```

Identify genes with high correlations to the RXLR genes listed.

```{r,corselect}

IUM83RxLR  <- readLines('IUMRxLR.csv')
IUM83RxLR <- IUM83RxLR[2:length(IUM83RxLR)]
str(IUM83RxLR)

length(IUM83RxLR)
IUM83RxLR <- IUM83RxLR[which(IUM83RxLR %in% rownames(rpm))]
length(IUM83RxLR)

corselect <- function(i) {
  gene <- IUM83RxLR[i]
  hist(genecor[,which(colnames(genecor)==gene)])
  mycors <- genecor[,which(colnames(genecor)==gene)]
  mytopcor <- mycors[which(mycors>0.8)]
  mytopcor <- mytopcor[order(-mytopcor)]
  return(mytopcor)
}

# test
corselect(2)

res <- lapply(X=1:length(IUM83RxLR),FUN=corselect)

names(res) <- IUM83RxLR

str(res)

lapply(res,function(vec) { head(vec,200) } )

```

```{r,pca1,fig.width=8,fig.height=8}

library(limma)
plotMDS(rpm)
#plotMDS(t(rpm))

```

## Session info

```{r,sessioninfo}

sessionInfo()


```
