---
title: "PcDESeq2"
author: "BWSchroeter"
date: "2023-05-22"
output:
  html_document: default
  word_document: default
---

```{r, load libraries, include=FALSE}
#library("kableExtra")
library(DESeq2)
library(GEOquery)
library(tidyverse)
library(gridExtra)
library(reshape2)
library(rtracklayer)
library(dplyr)
library(edgeR)
library(clusterProfiler)
library(pheatmap)
library(RColorBrewer)
library("eulerr")
library(officer)
library(flextable)


```

Upload files

```{r, include=FALSE}
setwd("../shared_scripts")

#import raw files
source("../shared_scripts/pcinnamomi_annotations.R", 
       local = knitr::knit_global())
source("../shared_scripts/sample_names.R", 
       local = knitr::knit_global())
source("../shared_scripts/countmatrix.R", 
       local = knitr::knit_global())

```

Clean coldata for Deseq2 normalisation

```{r}
colData <- coldata %>% 
  filter(str_detect(Sample, "Pc"))

rownames(colData) <- colData$Sample
colData <- colData [ ,-1] 

# making sure the row names in colData matches to column names in counts_data
all(colnames(IUM83CountMatrix) %in% rownames(colData))
```

```{r}
# Create sample information
sampleNames <- colData$Sample
condition <- colData$Stage
timepoint <- colData$Time

colDATA <- data.frame(sampleNames,
                      condition,
                      timepoint,
                      stringsAsFactors = T)

# Create the DESeqDataSet object
dds <- DESeqDataSetFromMatrix(countData = IUM83CountMatrix,
                              colData = colDATA,
                              design = ~ condition)

nrow(dds)
dds <- dds[rowMeans(counts(dds)) >10,]
nrow(dds)

dds <- estimateSizeFactors(dds)

```

```{r}
vsd <- vst(dds, blind = FALSE)
head(assay(vsd), 3)

rld <- rlog(dds, blind = FALSE)
head(assay(rld), 3)

```

```{r}
ds <- estimateSizeFactors(dds)

df <- bind_rows(
  as.data.frame(log2(counts(dds, normalized=F)[, 1:2]+1)) %>%
         mutate(transformation = "log2(x + 1)"),
  as.data.frame(assay(vsd)[, 1:2]) %>% mutate(transformation = "vst"),
  as.data.frame(assay(rld)[, 1:2]) %>% mutate(transformation = "rlog"))
  
colnames(df)[1:2] <- c("x", "y")  

lvls <- c("log2(x + 1)", "vst", "rlog")
df$transformation <- factor(df$transformation, levels=lvls)

ggplot(df, aes(x = x, y = y)) + geom_hex(bins = 80) +
  coord_fixed() + facet_grid( . ~ transformation) 
```

```{r}
sampleDists <- dist(t(assay(rld)))
sampleDists

sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste( vsd$dex, vsd$cell, sep = " - " )
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)

pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)

```

```{r}

PCA <- plotPCA(rld, c("timepoint", "sampleNames"), returnData = TRUE)
percentVar <- round(100 * attr(PCA, "percentVar"))

ggplot(PCA, aes(PC1, PC2, color = condition)) +
  geom_point(size = 5) +
  geom_point(shape = 21, fill = "transparent", colour = "black", size = 5) +
  xlab(paste0("PC1 (", percentVar[1], "%)")) +
  ylab(paste0("PC2 (", percentVar[2], "%)")) +
  coord_fixed() +
  theme_bw() +
  theme(axis.line = element_line(color = "black"),
        panel.border = element_rect(color = "black"),
        panel.grid.major = element_line(color = "grey"),  
        panel.grid.minor = element_blank(),
        legend.position = c(0.05, 0.40),  
        legend.justification = c(0, 1),  
        legend.background = element_rect(color = "black", fill = "white", size = 0.5),  
        legend.key.size = unit(1, "lines"))  

```

```{r}
sampleDists <- dist(t(assay(vsd)))
sampleDists

sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste( vsd$dex, vsd$cell, sep = " - " )
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)

```

```{r}
plotPCA(vsd, intgroup = "condition", returnData = FALSE)
```


```{r}

PCA <- plotPCA(vsd, c("timepoint", "sampleNames"), returnData = TRUE)
percentVar <- round(100 * attr(PCA, "percentVar"))

ggplot(PCA, aes(PC1, PC2, color = condition)) +
  geom_point(size = 5) +
  geom_point(shape = 21, fill = "transparent", colour = "black", size = 5) +
  xlab(paste0("PC1 (", percentVar[1], "%)")) +
  ylab(paste0("PC2 (", percentVar[2], "%)")) +
  coord_fixed() +
  theme_bw() +
  theme(axis.line = element_line(color = "black"),
        panel.border = element_rect(color = "black"),
        panel.grid.major = element_line(color = "grey"),  
        panel.grid.minor = element_blank(),
        legend.position = c(0.05, 0.95),  
        legend.justification = c(0, 1),  
        legend.background = element_rect(color = "black", fill = "white", size = 0.5), 
        legend.key.size = unit(1, "lines"))  

```

```{r}

png("../Figures/PCA_vst.png", width = 10, height = 6, units = "in", res = 300)

ggplot(PCA, aes(PC1, PC2, color = condition)) +
  geom_point(size = 5) +
  geom_point(shape = 21, fill = "transparent", colour = "black", size = 5) +
  xlab(paste0("PC1 (", percentVar[1], "%)")) +
  ylab(paste0("PC2 (", percentVar[2], "%)")) +
  coord_fixed() +
  theme_bw() +
  theme(axis.line = element_line(color = "black"),
        panel.border = element_rect(color = "black"),
        panel.grid.major = element_line(color = "grey"),  
        panel.grid.minor = element_blank(),
        legend.position = c(0.05, 0.95),  
        legend.justification = c(0, 1),  
        legend.background = element_rect(color = "black", fill = "white", size = 0.5),
        legend.key.size = unit(1, "lines"))  

dev.off()
```

```{r}
dds$condition <- relevel(dds$timepoint, ref = "0hpi")

dds <- DESeq(dds)

as.data.frame(colData(dds))
```

```{r, pheatmap annotation}
col_ann <- colData[,c(1,3)]
rownames(col_ann) <- col_ann$Sample
col_ann <- data.frame(col_ann)
col_ann$Stage <- as.factor(col_ann$Stage)
col_ann <- col_ann[order(col_ann$Stage),]
col_ann$sample_ID <- NULL

#remove samples to show only stage in figure
#col_ann$Sample <- NULL

head(col_ann, 12)

```

```{r}
# function to save tables landscape
sect_properties <- prop_section(
  page_size = page_size(
    orient = "landscape",
    width = 8.7, height = 11.7
  ),
  type = "continuous",
  page_margins = page_mar()
)
```

"It is therefore desirable to include the threshold in the statistical testing procedure directly, i.e., not to filter post hoc on a reported fold-change estimate, but rather to evaluate statistically directly whether there is sufficient evidence that the LFC is above the chosen threshold." - Love, M.I., Huber, W. & Anders, S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol 15, 550 (2014). https://doi.org/10.1186/s13059-014-0550-8


```{r, Early summary}

res_early_vs_vegetative <- results(dds, alpha=0.05, lfcThreshold = 1, contrast = c("condition", "18hpi", "0hpi"))
summary(res_early_vs_vegetative)
(res_early_vs_vegetative)

```

```{r, Top 50 most significant genes based on p-value - Early}
top_genes_early <- head(res_early_vs_vegetative[order(res_early_vs_vegetative$pvalue), ], 50)

top_gene_expression_early.id <- as.data.frame(top_genes_early)
top_gene_expression_early.id$gene_id <- rownames(top_gene_expression_early.id)
top_gene_expression_early.id <- dplyr::right_join(gtf.omicsbox, top_gene_expression_early.id, by = 'gene_id') %>%
  arrange(padj)

# Extract the expression values for the top genes
top_gene_expression_early <- assay(dds)[rownames(top_genes_early), ]

# Extract the expression values for the top genes and select columns 1 to 6
top_gene_expression_early <- top_gene_expression_early[, 1:6]

# Create a matrix of log2-transformed expression values
#log2_expression_early <- log2(top_gene_expression_early + 1)

png(filename = "../Figures/Earlynormscaled.png", width = 3600, height = 6400, res = 600)

deph1 <- pheatmap(top_gene_expression_early, 
         scale = "row",
         cluster_rows = TRUE,
         treeheight_row = 0,
         fontsize_number = 16, 
         annotation_col = col_ann,
         show_colnames = TRUE,
         show_rownames = FALSE,
         fontsize_col = 14,
         cellheight = 10, cellwidth = 40,
         main = "Early interaction")

dev.off()

```

```{r, Early volcano basemean}
# early DE 
ede <- as.data.frame(res_early_vs_vegetative[order(res_early_vs_vegetative$padj), ])
# early significant 

edesig <- subset(ede, padj < 0.05)

top_genes_ede <- edesig %>%
  dplyr::filter(baseMean >= 1000) %>%
  arrange(pvalue) %>%
  head(50)

edeup_dn <- subset(ede, (log2FoldChange > 1 | log2FoldChange < -1) & padj < 0.05)

edeup <- rownames(subset(ede, log2FoldChange > 1 & padj < 0.05))
ededn <- rownames(subset(ede, log2FoldChange < -1 & padj < 0.05))
# Calculate the gene counts
EDEUP <- length(edeup)
EDEDN <- length(ededn)

SUBHEADER <- paste(EDEUP, "Up-regulated genes ,", EDEDN, " Down-regulated genes")

evol_df <- data.frame(
  padj = top_genes_ede$padj,
  log2FoldChange = top_genes_ede$log2FoldChange,
  baseMean = top_genes_ede$baseMean,
  gene_id = rownames(top_genes_ede)
  )
etop_genes <- head(evol_df[order(evol_df$padj), ], 10) %>%
  mutate(Rank = row_number())

png(filename = "../Figures/EarlyDEvolcano2.png", width = 4800, height = 4800, pointsize = 10, res = 600)

ggplot(evol_df, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(color = "black", alpha = 0.5, size = 2, shape = 19) +
  geom_point(data = ede, aes(color = "detected genes"), size = 2, shape = 19) +
  geom_point(data = edeup_dn, aes(color = "significantly expressed genes"), size = 2, shape = 19) +
  geom_text(data = subset(etop_genes), aes(label = Rank),
         hjust = 0.5, vjust = 0.5, color = "black", size = 3) +
         labs(x = "log2foldchange", y = "log10 p-value") +
  scale_color_manual(values = c("significantly expressed genes" = "turquoise", "detected genes" = "grey"),
                     name = "") +
  theme_bw() +
  theme(axis.line = element_line(color = "black"),
        panel.border = element_rect(color = "black"),
        panel.grid.major = element_line(color = "grey"),  
        panel.grid.minor = element_blank(),
        legend.position = c(0.05, 0.99),  
        legend.justification = c(0, 1),  
        legend.background = element_rect(fill = NA, size = 0.5),
        legend.key.size = unit(1, "lines")) +
  ggtitle(expression(paste(italic("Phytophthora cinnamomi"), " gene expression during early interaction"))) +
  labs(subtitle = SUBHEADER) +
  theme(plot.title = element_text(face = "bold", size = 16)) 

dev.off()
```

```{r}
earlydeseqtable <- dplyr::right_join(gtf.omicsbox, etop_genes, by = 'gene_id') %>%
  arrange(Rank)

earlydeseqtable <- head(earlydeseqtable, 10)

# Order columns logically 
earlydeseqtable <- earlydeseqtable %>% 
  dplyr::select(c(Rank, gene_id, protein_id, product, GO_terms, baseMean, log2FoldChange, padj))

earlydeseqtable$baseMean <- round(earlydeseqtable$baseMean)
earlydeseqtable$log2FoldChange <- round(earlydeseqtable$log2FoldChange, 2)
earlydeseqtable$padj <- as.numeric(earlydeseqtable$padj)
# Format the 'padj' column to scientific notation with two decimal places
earlydeseqtable$padj <- sprintf("%.2e", earlydeseqtable$padj)

# Rename columns 
colnames(earlydeseqtable) <- c("Rank", "Gene name", "Protein name", "Product", "GO term", "baseMean", "log2FoldChange", "padj")

earlydeseqtable <-  flextable(earlydeseqtable) %>%
  theme_booktabs() %>%
  align(align = "center", part = c("all"))

print(earlydeseqtable)

#save_as_docx(earlydeseqtable, path =  "/mnt/data/barry/projects/Schroeter_Pcinnamomi/sequencing_stats/earlydeseqtable.docx", pr_section = sect_properties)
```

```{r, Middle summary}
# To compare "Middle" vs "Vegetative"
res_middle_vs_vegetative <- results(dds, alpha=0.05, lfcThreshold = 1, contrast =c ("condition", "30hpi", "0hpi"))
summary(res_middle_vs_vegetative)
res_middle_vs_vegetative
```

```{r, Top 50 most significant genes based on p-value - Middle}

top_genes_mid <- head(res_middle_vs_vegetative[order(res_middle_vs_vegetative$pvalue), ], 50)

top_gene_expression_mid.id <- as.data.frame(top_genes_mid)
top_gene_expression_mid.id$gene_id <- rownames(top_gene_expression_mid.id)
top_gene_expression_mid.id <- dplyr::right_join(gtf.omicsbox, top_gene_expression_mid.id, by = 'gene_id') %>%
  arrange(padj)

# Extract the expression values for the top genes
top_gene_expression <- assay(dds)[rownames(top_genes_mid), ]

# Extract the expression values for the top genes and the selected columns (1 to 3 and 7 to 9)
top_gene_expression_mid <- assay(dds)[rownames(top_genes_mid), c(1:3, 7:9)]

# Create a matrix of log2-transformed expression values
#log2_expression_mid <- log2(top_gene_expression_mid + 1)

png(filename = "../Figures/Midnormscaled.png", width = 3600, height = 6400, res = 600)

deph2 <- pheatmap(top_gene_expression_mid,
         scale = "row",
         cluster_rows = TRUE,
         treeheight_row = 0,
         fontsize_number = 16, 
         annotation_col = col_ann,
         show_colnames = TRUE,
         show_rownames = FALSE,
         fontsize_col = 14,
         cellheight = 10, cellwidth = 40,
         main = "Mid interaction"
         )

dev.off()
```

```{r, middle volcano}
# Mid DE 
mde <- as.data.frame(res_middle_vs_vegetative[order(res_middle_vs_vegetative$padj), ])
# Mid significant 
mdesig <- subset(mde, padj < 0.05)

top_genes_mde <- mdesig %>%
  dplyr::filter(baseMean >= 1000) %>%
  arrange(pvalue) %>%
  head(50)

mdeup_dn <- subset(mde, (log2FoldChange > 1 | log2FoldChange < -1) & padj < 0.05)

mdeup <- rownames(subset(mde, log2FoldChange > 1 & padj < 0.05))
mdedn <- rownames(subset(mde, log2FoldChange < -1 & padj < 0.05))
# Calculate the gene counts
MDEUP <- length(mdeup)
MDEDN <- length(mdedn)

midgenesDE <- head(top_gene_expression_mid.id[order(top_gene_expression_mid.id$pvalue), ], 10)
midgenesDE <- midgenesDE$gene_id
# Create the subtitle using the gene counts
SUBHEADER <- paste(MDEUP, "Up-regulated genes ,", MDEDN, " Down-regulated genes")

mvol_df <- data.frame(
  padj = top_genes_mde$padj,
  log2FoldChange = top_genes_mde$log2FoldChange,
  baseMean = top_genes_mde$baseMean,
  gene_id = rownames(top_genes_mde)
  )
mtop_genes <- head(mvol_df[order(mvol_df$padj), ], 10) %>%
  mutate(Rank = row_number())

png(filename = "../Figures/MidDEvolcano.png", width = 4800, height = 4800, pointsize = 10, res = 600)

ggplot(mvol_df, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(color = "black", alpha = 0.5, size = 2, shape = 19) +
  geom_point(data = mde, aes(color = "detected genes"), size = 2, shape = 19) +
  geom_point(data = mdeup_dn, aes(color = "significantly expressed genes"), size = 2, shape = 19) +
  geom_text(data = subset(mtop_genes, baseMean > 1000), aes(label = Rank),
         hjust = 0.5, vjust = 0.5, color = "black", size = 3) +
         labs(x = "log2foldchange", y = "log10 p-value") +
  scale_color_manual(values = c("significantly expressed genes" = "turquoise", "detected genes" = "grey"),
                     name = "") +
  theme_bw() +
  theme(axis.line = element_line(color = "black"),
        panel.border = element_rect(color = "black"),
        panel.grid.major = element_line(color = "grey"),  
        panel.grid.minor = element_blank(),
        legend.position = c(0.05, 0.99),  
        legend.justification = c(0, 1),  
        legend.background = element_rect(fill = NA, size = 0.5),
        legend.key.size = unit(1, "lines")) +
  ggtitle(expression(paste(italic("Phytophthora cinnamomi"), " gene expression during mid interaction"))) +
  labs(subtitle = SUBHEADER) +
  theme(plot.title = element_text(face = "bold", size = 16)) 

dev.off()
```

```{r}
middeseqtable <- dplyr::right_join(gtf.omicsbox, mtop_genes, by = 'gene_id') %>%
  arrange(Rank)

middeseqtable <- head(middeseqtable, 10)

# Order columns logically 
middeseqtable <- middeseqtable %>% 
  dplyr::select(c(Rank, gene_id, protein_id, product, GO_terms, baseMean, log2FoldChange, padj))

middeseqtable$baseMean <- round(middeseqtable$baseMean)
middeseqtable$log2FoldChange <- round(middeseqtable$log2FoldChange, 2)
middeseqtable$padj <- as.numeric(middeseqtable$padj)
# Format the 'padj' column to scientific notation with two decimal places
middeseqtable$padj <- sprintf("%.2e", middeseqtable$padj)

# Rename columns 
colnames(middeseqtable) <- c("Rank","Gene name", "Protein name", "Product", "GO term", "baseMean", "log2FoldChange", "padj")

middeseqtable <-  flextable(middeseqtable) %>%
  theme_booktabs() %>%
  align(align = "center", part = c("all"))

print(middeseqtable)

#save_as_docx(middeseqtable, path =  "/mnt/data/barry/projects/Schroeter_Pcinnamomi/sequencing_stats/middeseqtable.docx", pr_section = sect_properties)
```

```{r, late summary}
# To compare "Late" vs "Vegetative"
res_late_vs_vegetative <- results(dds, alpha=0.05, lfcThreshold = 1, contrast = c("condition", "48hpi", "0hpi"))
summary(res_late_vs_vegetative)
(res_late_vs_vegetative)
```

```{r, late top 50 most significant genes based on p-value}
top_genes_late <- head(res_late_vs_vegetative[order(res_late_vs_vegetative$pvalue), ], 50)

top_gene_expression_late.id <- as.data.frame(top_genes_late)
top_gene_expression_late.id$gene_id <- rownames(top_gene_expression_late.id)
top_gene_expression_late.id <- dplyr::right_join(gtf.omicsbox, top_gene_expression_late.id, by = 'gene_id') %>%
  arrange(padj)

# Extract the expression values for the top genes and the selected columns (1 to 3 and 10 to 12)
top_gene_expression_late <- assay(dds)[rownames(top_genes_late), c(1:3, 10:12)]

# Create a matrix of log2-transformed expression values
log2_expression_late <- log2(top_gene_expression_late + 1)

png(filename = "../Figures/Latenormscaled.png", width = 3600, height = 6400, res = 600)

deph3 <- pheatmap(log2_expression_late,
         scale = "row",
         cluster_rows = TRUE,
         treeheight_row = 0,
         fontsize_number = 16, 
         annotation_col = col_ann,
         show_colnames = TRUE,
         show_rownames = FALSE,
         fontsize_col = 14,
         cellheight = 10, cellwidth = 40,
         main = "Late interaction")
       
dev.off()
```

```{r, late volcano}
# late DE 
lde <- as.data.frame(res_late_vs_vegetative[order(res_late_vs_vegetative$padj), ])
# late significant 
ldesig <- subset(lde, padj < 0.05)

top_genes_lde <- ldesig %>%
  dplyr::filter(baseMean >= 1000) %>%
  arrange(pvalue) %>%
  head(50)

ldeup_dn <- subset(lde, (log2FoldChange > 1 | log2FoldChange < -1) & padj < 0.05)

ldeup <- rownames(subset(lde, log2FoldChange > 1 & padj < 0.05))
ldedn <- rownames(subset(lde, log2FoldChange < -1 & padj < 0.05))
# Calculate the gene counts
LDEUP <- length(ldeup)
LDEDN <- length(ldedn)

lategenesDE <- head(top_gene_expression_late.id[order(top_gene_expression_late.id$pvalue), ], 10)
lategenesDE <- lategenesDE$gene_id

# Create the subtitle using the gene counts
SUBHEADER <- paste(LDEUP, "Up-regulated genes ,", LDEDN, " Down-regulated genes")

lvol_df <- data.frame(
  padj = top_genes_lde$padj,
  log2FoldChange = top_genes_lde$log2FoldChange,
  baseMean = top_genes_lde$baseMean,
  gene_id = rownames(top_genes_lde)
  )

ltop_genes <- head(lvol_df[order(lvol_df$padj), ], 10) %>%
  mutate(Rank = row_number())

png(filename = "../Figures/LateDEvolcano.png", width = 4800, height = 4800, pointsize = 10, res = 600)

ggplot(lvol_df, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(color = "black", alpha = 0.5, size = 2, shape = 19) +
  geom_point(data = lde, aes(color = "detected genes"), size = 2, shape = 19) +
  geom_point(data = ldeup_dn, aes(color = "significantly expressed genes"), size = 2, shape = 19) +
  geom_text(data = subset(ltop_genes, baseMean > 1000), aes(label = Rank),
         hjust = 0.5, vjust = 0.5, color = "black", size = 3) +
         labs(x = "log2foldchange", y = "log10 p-value") +
  scale_color_manual(values = c("significantly expressed genes" = "turquoise", "detected genes" = "grey"),
                     name = "") +
  theme_bw() +
  theme(axis.line = element_line(color = "black"),
        panel.border = element_rect(color = "black"),
        panel.grid.major = element_line(color = "grey"),  
        panel.grid.minor = element_blank(),
        legend.position = c(0.05, 0.99),  
        legend.justification = c(0, 1),  
        legend.background = element_rect(fill = NA, size = 0.5),
        legend.key.size = unit(1, "lines")) +
  ggtitle(expression(paste(italic("Phytophthora cinnamomi"), " gene expression during late interaction"))) +
  labs(subtitle = SUBHEADER) +
  theme(plot.title = element_text(face = "bold", size = 16)) 

dev.off()
```

```{r}
latedeqtable <- dplyr::right_join(gtf.omicsbox, ltop_genes, by = 'gene_id') %>%
  arrange(Rank)

# Order columns logically 
latedeqtable <- latedeqtable %>% 
  dplyr::select(c(Rank, gene_id, protein_id, product, GO_terms, baseMean, log2FoldChange, padj))

latedeqtable$baseMean <- round(latedeqtable$baseMean)
latedeqtable$log2FoldChange <- round(latedeqtable$log2FoldChange, 2)
latedeqtable$padj <- as.numeric(latedeqtable$padj)
# Format the 'padj' column to scientific notation with two decimal places
latedeqtable$padj <- sprintf("%.2e", latedeqtable$padj)

# Rename columns 
colnames(latedeqtable) <- c("Rank", "Gene name", "Protein name", "Product", "GO term", "baseMean", "log2FoldChange", "padj")

latedeqtable <-  flextable(latedeqtable) %>%
  theme_booktabs() %>%
  align(align = "center", part = c("all"))

print(latedeqtable)

#save_as_docx(latedeqtable, path =  "/mnt/data/barry/projects/Schroeter_Pcinnamomi/sequencing_stats/latedeseqtable.docx", pr_section = sect_properties)

```

```{r}
deseqtable <- read.csv("/mnt/data/barry/projects/Schroeter_Pcinnamomi/shared_scripts/finaltable.tsv")

colnames(deseqtable) <- c("gene_id", "Protein name", "Product", "GO BP terms", "GO BP IDs", "GO MF terms", "GO MF IDs", "GO CC terms", "GO CC IDs", "SP, TM", "Effector")

```

```{r, differentially expressed genes, echo=FALSE}

detectedde1 <- rownames(subset(ede, (log2FoldChange > 1 | log2FoldChange < -1) & padj < 0.05))
detectedde2 <- rownames(subset(mde, (log2FoldChange > 1 | log2FoldChange < -1) & padj < 0.05))
detectedde3 <- rownames(subset(lde, (log2FoldChange > 1 | log2FoldChange < -1) & padj < 0.05))

combined_devector <- c(detectedde1, detectedde2, detectedde3)

# Find the unique elements
uniquede_elements <- unique(combined_devector)

# Number of unique elements
uniquede_count <- length(uniquede_elements)

```

```{r, de expression eulerr plot, echo=FALSE}

png("upregulated.png", width = 10, height = 6, units = "in", res = 300)

upregulated <- list("Early-up"=edeup, "Mid-up"=mdeup,"Late-up"=ldeup)

plot(euler(upregulated),quantities = TRUE, labels = list(font = 24), main="Upregulated")

dev.off()

unique_in_edeup <- setdiff(edeup, union(mdeup, ldeup))
unique_in_mdeup <- setdiff(mdeup, union(edeup, ldeup))
unique_in_ldeup <- setdiff(ldeup, union(edeup, mdeup))

upfiltered_gtf <- deseqtable %>%
  filter(gene_id %in% unique_in_ldeup)

```

```{r, de expression eulerr plot, echo=FALSE}

png("downregulated.png", width = 10, height = 6, units = "in", res = 300)

downregulated <- list("Early-down"=ededn, "Mid-down"=mdedn, "Late-down"=ldedn)

plot(euler(downregulated),quantities = TRUE, labels = list(font = 24), main="Downregulated")

unique_in_ededn <- setdiff(ededn, union(mdedn, ldedn))
unique_in_mdedn <- setdiff(mdedn, union(ededn, ldedn))
unique_in_ldedn <- setdiff(ldedn, union(ededn, mdedn))

dnfiltered_gtf <- gtf.omicsbox %>%
  filter(gene_id %in% unique_in_mdedn)

dev.off()
```


## Session information

```{r,session}

sessionInfo()

```
