```{r, libs, message=FALSE}
library(readxl)
library("kableExtra")
library(tidyverse)
library(gridExtra)
library(reshape2)
library(dplyr)
library(edgeR)
library(clusterProfiler)
library(BiocGenerics)
library(GenomicFeatures)
library(Biostrings)
library(rtracklayer)
library(data.table)
```

download .faa file https://ftp.ncbi.nlm.nih.gov/genomes/genbank/protozoa/Phytophthora_cinnamomi/latest_assembly_versions/GCA_018691715.1_ASM1869171v1/GCA_018691715.1_ASM1869171v1_protein.faa.gz

```{r}
#setwd("~/projects/Schroeter_Pcinnamomi/PcEffector")

# Read the .faa file
faa_file <- "GCA_018691715.1_ASM1869171v1_protein.faa"  
sequences <- readAAStringSet(faa_file, format = "fasta")

formatted_sequences <- paste0(">", names(sequences), "\n", as.character(sequences), "\n")
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Pcgenes.fasta" 
writeLines(formatted_sequences, output_file)


load("Pceffetor.RData")

# first rearrange cols
allgenes <- detectedgenes
allgenes$gene_id<-rownames(allgenes)
rownames(allgenes)<-seq(nrow(allgenes))
head(allgenes)


Pcgenenames <- read.table('pcgeneids.tsv', row.names = 1, quote = "", sep='\t', fill = TRUE, header = TRUE)

pcprotein.id <- dplyr::right_join(Pcgenenames, allgenes, by = 'gene_id')


# Define a list of protein IDs to extract
protein_list <- pcprotein.id$protein_id

```

```{r}
# Read the .faa file
faa_file <- "GCA_018691715.1_ASM1869171v1_protein.faa"  
sequences <- readAAStringSet(faa_file)

# Extract headers and sequences
headers <- names(sequences)
sequence_strings <- as.character(sequences)

# Extract protein identifiers from headers
protein_identifiers <- sub("^>(\\S+).*", "\\1", headers)

# Create a data frame with protein identifiers and sequences
df <- data.frame(protein_id = protein_identifiers, Sequence = sequence_strings, stringsAsFactors = FALSE)

# Filter the "ProteinID" column to include only the first 12 characters
df$protein_id <- substr(df$protein_id, 1, 12)

# Print the updated data frame
head(df)

pcprotein.idfaa <- dplyr::right_join(df, pcprotein.id, by = 'protein_id')
pcprotein.idfaa <- pcprotein.idfaa[, 1:4]
head(pcprotein.idfaa)
```
function used to create .fasta for inputs

```{r}
#function used to make .fasta files for analysis
write_fasta <- function(df, file_path) {
  fasta_string <- paste0(">", df$protein_id, "\n", df$Sequence, "\n")
  writeLines(fasta_string, con = file_path)
}

generate_fasta_file <- function(df, output_file) {
  write_fasta(df, output_file)
}
```

```{r}
testfasta <- head(pcprotein.idfaa)
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/test.fasta"

generate_fasta_file(testfasta, output_file)
#use for testing terminal scripts (InterProScan, SignalP and EffectorP) - 6 genes
```

```{r}
#Pc genes detect in transcriptome -- 13595 genes detected 
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/detected_genes.fasta"

generate_fasta_file(pcprotein.idfaa, output_file)

```

```{r}
#all Pc genes from .faa file -- 21335 genes 
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta"

generate_fasta_file(df, output_file)

```

###
InterProScan analysis
###
barry@ziemann-01:/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan/interproscan-5.62-94.0$ ./interproscan.sh -appl CDD,PANTHER,Pfam,PROSITEPATTERNS,PROSITEPROFILES,SFLD,SMART,TIGRFAM,Phobius,TMHMM,SUPERFAMILY -i /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta -iprlookup --goterms -b /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan_output/
05/06/2023 12:56:46:576 Welcome to InterProScan-5.62-94.0
05/06/2023 12:56:46:577 Running InterProScan v5 in STANDALONE mode... on Linux
05/06/2023 12:56:51:785 RunID: ziemann-01_20230605_125651556_b2fz
05/06/2023 12:57:02:847 Loading file /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta
05/06/2023 12:57:02:848 Running the following analyses:
[CDD-3.20,NCBIfam-11.0,PANTHER-17.0,Pfam-35.0,Phobius-1.01,ProSitePatterns-2022_05,ProSiteProfiles-2022_05,SFLD-4,SMART-9.0,SUPERFAMILY-1.75,TMHMM-2.0c]
Available matches will be retrieved from the pre-calculated match lookup service.

Matches for any sequences that are not represented in the lookup service will be calculated locally.
05/06/2023 12:57:07:851 Uploaded 20850 unique sequences for analysis
05/06/2023 12:59:50:492 87% completed
05/06/2023 13:59:50:678 19% completed
 05/06/2023 14:59:50:824 24% completed
05/06/2023 15:59:51:007 77% completed
05/06/2023 16:12:19:464 100% done:  InterProScan analyses completed 

```{r}
interproscanresults <- read_tsv("allpcgenes.fasta.tsv")

colnames(interproscanresults) <- c("protein_id", "protein_md5", "protein_length", 
                     "analysis", "signature_id", 
                     "signature_desc", "start", "end", 
                     "score", "status", "date", 
                     "interpro_id", "interpro_desc", "go_terms")
head(interproscanresults)

uniqueips <- distinct(interproscanresults, protein_id)

```

```{r}
#extract signal peptides from interproscan results 
interproscanSP <- unique(interproscanresults[interproscanresults$signature_id == "SIGNAL_PEPTIDE",])

#3050 proteins with predicted signal peptides from 16957 proteins scanned - less than 80 amino acids not scanned by interProScan
```

###
SignalP analysis
###

setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/signalp-6.0/signalp6_fast/")
Sys.setenv(PATH = paste("/home/barry/.local/bin", Sys.getenv("PATH"), sep = ":"))

SIGNALP_DIR <- system("python3 -c 'import signalp; import os; print(os.path.dirname(signalp.__file__))'", intern = TRUE)

Sys.getenv("PATH")

system("signalp6 -h")

system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")


system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")
Predicting: 100%|██████████| 21334/21334 [39:18<00:00,  9.05sequences/s]

# run on signalP_6.0 and import results 

```{r}
# Read the prediction_results.txt file into a data frame
prediction_results <- read.table("prediction_results.txt", header = F, comment.char = "#", sep = "\t")
colnames(prediction_results) <- c("protein_id", "Prediction", "OTHER", "SP(Sec/SPI)", "CS Position")
head(prediction_results)

# Filter the prediction_results data frame by "SP" in the Prediction column
signalPsecreted <- filter(prediction_results, Prediction == "SP")

# Use separate to split CS Position column
split_results <- signalPsecreted %>%
  separate(`CS Position`, into = c("cleavage_site", "probability"),
           sep = "\\. ", remove = FALSE)

signalPSP <- split_results %>%
  mutate(cleavage_site = gsub("CS pos: ", "", cleavage_site),
         probability = gsub("Pr: ", "", probability))

head(signalPSP)

#1878/21334 proteins predicted to contain a signal peptide using signalP6.0 

split_results$probability <- as.numeric(signalPSP$probability)
hist(split_results$probability, main = "Probability Histogram", xlab = "Probability")

#filter out low probability SP
signalPSP.filtered <- subset(signalPSP, probability >= 0.7)

#1694/1878 with a probability of 0.7 or greater

```

```{r}
#minus those that are OTHER in signalP (removes some ribosomal proteins)
signalPnon <- filter(prediction_results, Prediction == "OTHER")
# Find the common protein IDs
common_protein_ids <- intersect(interproscanSP$protein_id, signalPnon$protein_id)

# Remove the common protein IDs from the dataset
interproscanSP <- interproscanSP[!interproscanSP$protein_id %in% common_protein_ids, ]

```


```{r}
#join interproscan and signalP results together use this file for effectorP analysis
interproscan_protein_ids <- interproscanSP$protein_id
signalP_protein_ids <- signalPSP$protein_id

combined_protein_ids <- c(interproscan_protein_ids, signalP_protein_ids)

totalSP <- (unique(combined_protein_ids))

SPdf <- df[df$protein_id %in% totalSP, ]

#1878 genes predicted to have signal peptides
```

###
Known Pc effectors 
###

RxLR

Get supplementary files from https://rdcu.be/dcczO
https://static-content.springer.com/esm/art%3A10.1186%2Fs12864-021-07552-y/MediaObjects/12864_2021_7552_MOESM2_ESM.xls

```{r, Engelbrecht supp}
#only considered the 181 candidate RxLR -IUM83_06578-RA/RB alternatively spliced total 180 rxlr

pcrxlr <- read_xls('12864_2021_7552_MOESM2_ESM.xls', 
                   sheet = 'TableS2',
                   col_names = F,
                   range = 'A3:A183') 
# Rename the column
colnames(pcrxlr) <- c("gene_id")

# Remove the last three characters of each row
pcrxlr$gene_id <- substr(pcrxlr$gene_id, 1, nchar(pcrxlr$gene_id) - 3)

as.data.frame(pcrxlr)

rxlr.df <- Pcgenenames %>%
  filter(gene_id %in% pcrxlr$gene_id)

head(rxlr.df)
```

```{r}
#49 CRNs (45 genes, of which four have alternative transcript variants)

pccrn <- read_xls('12864_2021_7552_MOESM2_ESM.xls', 
                   sheet = 'TableS3',
                   col_names = F,
                   range = 'A3:A51') 
# Rename the column
colnames(pccrn) <- c("gene_id")

# Remove the last three characters of each row
pccrn$gene_id <- substr(pccrn$gene_id, 1, nchar(pccrn$gene_id) - 3)

as.data.frame(pccrn)

crn.df <- Pcgenenames %>%
  filter(gene_id %in% pccrn$gene_id)

head(crn.df)
```

```{r}
#61 necrosis-inducing proteins (NLPs)

pcnlp <- read_xls('12864_2021_7552_MOESM2_ESM.xls', 
                   sheet = 'TableS4',
                   col_names = F,
                   range = 'A3:A63') 
# Rename the column
colnames(pcnlp) <- c("gene_id")

# Remove the last three characters of each row
pcnlp$gene_id <- substr(pcnlp$gene_id, 1, nchar(pcnlp$gene_id) - 3)

as.data.frame(pcnlp)

nlp.df <- Pcgenenames %>%
  filter(gene_id %in% pcnlp$gene_id)

head(nlp.df)
```

```{r}
#list of know effectors based on published data 
knowneffectors <- as.data.frame(c(nlp.df$protein_id, crn.df$protein_id, rxlr.df$protein_id))
colnames(knowneffectors)[colnames(knowneffectors) == "c(nlp.df$protein_id, crn.df$protein_id, rxlr.df$protein_id)"] <- "protein_id"
#add known effectors to predicted effectors

allsecreted <- unique(c(nlp.df$protein_id, crn.df$protein_id, rxlr.df$protein_id, SPdf$protein_id))

allsecreted <- as.data.frame(allsecreted)
names(allsecreted) <- "protein_id"

allsecreted.id <- dplyr::right_join(df, allsecreted, by = 'protein_id')

output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector//EffectorP_3.0.0-beta/allsecreted.fasta"

generate_fasta_file(allsecreted.id, output_file)

detectedsecreted.id <- dplyr::right_join(pcprotein.idfaa, allsecreted, by = 'protein_id')
```

###
EffectorP analysis
###

```{r}
#path to where the 
setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/EffectorP_3.0.0-beta/")

system("/home/barry/.virtualenvs/my-python/bin/python EffectorP.py -i allsecreted.fasta > allsecreted_effectorPresults.txt")

```

-----------------
1878 proteins were provided as input in the following file: allSP.fasta
-----------------
Number of predicted effectors: 741
Number of predicted cytoplasmic effectors: 379
Number of predicted apoplastic effectors: 362
-----------------
39.5 percent are predicted effectors.
20.2 percent are predicted cytoplasmic effectors.
19.3 percent are predicted apoplastic effectors.
-----------------
6.3 percent of the cytoplasmic effectors are predicted dual-localized (cytoplasmic/apoplastic).
11.3 percent of the apoplastic effectors are predicted dual-localized (apoplastic/cytoplasmic).
-----------------
-----------------

```{r}
effectorp.results <- read.delim("allSP_effectorPresults.txt", sep = "\t", stringsAsFactors = FALSE, skip = 10)
effectorp.results <- effectorp.results[1:(nrow(effectorp.results) - 15), ]
colnames(effectorp.results) <- c("protein_id", "Cytoplasmic_effector", "Apoplastic_effector", "Non_effector", "Prediction")
effectorp.results$protein_id <- gsub(" ", "", effectorp.results$protein_id)

head(effectorp.results)

detectedeffectors <- dplyr::right_join(pcprotein.idfaa, effectorp.results, by = 'protein_id')

```



```{r}
#save .RDS including predicted secreted, known effectors, and  to use in PcDESeq2.rmd

#save(allsecreted.id, detectedeffectors, knowneffectors, file = "secrete.effect.RData")

```

library("eulerr") 
v1 <- list("edgeR up"=dge_edger_up, "edgeR dn"=dge_edger_dn,  "DESeq2 up"=dge_deseq2_up,"DESeq2 dn"=dge_deseq2_dn)plot(euler(v1),quantities = TRUE)
library("eulerr")v1 <- list("vec1"=vec1, "vec2"=vec2,"vec3"=vec3)plot(euler(v1),quantities = TRUE)

intersect(vec1,vec2)




###
bdCAN analysis
###

https://bcb.unl.edu/dbCAN2/blast.php using allpcgenes.fasta

check all boxes except substrate prediction
HMMER: dbCAN (E-Value < 1e-15, coverage > 0.35)
DIAMOND: CAZy (E-Value < 1e-102
HMMER: dbCAN-sub (E-Value < 1e-15, coverage > 0.35)

Jinfang Zheng and others, dbCAN3: automated carbohydrate-active enzyme and substrate annotation, 
Nucleic Acids Research, 2023;, gkad328, https://doi.org/10.1093/nar/gkad328



```{r}
dbcanresults <- read_tsv("dbCAN_allpcgenes.txt")
head(dbcanresults)

dbcan <- dbcanresults %>%
  filter(`#ofTools` >= 2)

print(dbcan)

#435 predicted CAZymes version dbCAN3.0
#paper predicted 468 they had HMMER/Hotpep/DIAMOND  Hotpep replaced with eCAMI obsolete version dbCAN2.0
```

