
```{r, libs, message=FALSE}
library("kableExtra")
library(tidyverse)
library(gridExtra)
library(reshape2)
library(dplyr)
library(edgeR)
library("eulerr")
library(data.table)
```

```{r, include=FALSE}
source("../shared_scripts/pcinnamomi_annotations.R", 
       local = knitr::knit_global())
source("../shared_scripts/sample_names.R", 
       local = knitr::knit_global())
source("../shared_scripts/countmatrix.R", 
       local = knitr::knit_global())

```

```{r}
#function used to make .fasta files for analysis
write_fasta <- function(df, file_path) {
  fasta_string <- paste0(">", df$protein_id, "\n", df$Sequence, "\n")
  writeLines(fasta_string, con = file_path)
}

generate_fasta_file <- function(df, output_file) {
  write_fasta(df, output_file)
}
```

```{r}
#all Pc genes from .faa file -- 21334 genes 
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta"

generate_fasta_file(df, output_file)

```

#######################
 InterProScan analysis
#######################

barry@ziemann-01:/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan/interproscan-5.62-94.0$ ./interproscan.sh -appl CDD,PANTHER,Pfam,PROSITEPATTERNS,PROSITEPROFILES,SFLD,SMART,TIGRFAM,Phobius,TMHMM,SUPERFAMILY -i /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta -iprlookup --goterms -b /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan_output/
05/06/2023 12:56:46:576 Welcome to InterProScan-5.62-94.0
05/06/2023 12:56:46:577 Running InterProScan v5 in STANDALONE mode... on Linux
05/06/2023 12:56:51:785 RunID: ziemann-01_20230605_125651556_b2fz
05/06/2023 12:57:02:847 Loading file /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta
05/06/2023 12:57:02:848 Running the following analyses:
[CDD-3.20,NCBIfam-11.0,PANTHER-17.0,Pfam-35.0,Phobius-1.01,ProSitePatterns-2022_05,ProSiteProfiles-2022_05,SFLD-4,SMART-9.0,SUPERFAMILY-1.75,TMHMM-2.0c]
Available matches will be retrieved from the pre-calculated match lookup service.

Matches for any sequences that are not represented in the lookup service will be calculated locally.
05/06/2023 12:57:07:851 Uploaded 20850 unique sequences for analysis
05/06/2023 12:59:50:492 87% completed
05/06/2023 13:59:50:678 19% completed
 05/06/2023 14:59:50:824 24% completed
05/06/2023 15:59:51:007 77% completed
05/06/2023 16:12:19:464 100% done:  InterProScan analyses completed 

```{r}
source("../shared_scripts/effectors_from_literature.R", 
       local = knitr::knit_global())

```

```{r}
#extract signal peptides from interproscan results 
interproscanSP <- interproscanresults[interproscanresults$signature_id == "SIGNAL_PEPTIDE",] 
interproscanSP <- unique(interproscanSP$protein_id)

phobius.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% interproscanSP, ]
#Phobius predicts 3050 proteins with predicted signal peptides from 16957 proteins scanned - less than 80 amino acids not scanned by interProScan introduces ribosomal proteins - look at removing them based on er retention signals? further work. Although may be a nuclear signal that is seen in plants that target from chloroplast to nucleus

#3050 proteins
```

```{r}
interproscanTMHMM <- unique(interproscanresults[interproscanresults$signature_id == "TMhelix",])

filtered_interproscanTMHMM <- interproscanTMHMM %>%
  group_by(protein_id) %>%
  filter(n() <= 1) %>% 
  filter(end <= 60)

filtered_interproscanTMHMM.id <- dplyr::right_join(gtf.faa.omicsbox, filtered_interproscanTMHMM, by = 'protein_id')

#TMHMM predicts 594 proteins that aren't considered transmembrane
```

Tmhmm v.2.0 was used to predict transmembrane proteins from the combined predicted secretome (Krogh et al., 2001). If a protein contains at least one transmembrane domain after the first 60 amino acids or at least two transmembrane domains in total, it was labelled as a putative transmembrane protein.

Ranking effector candidates according to their expression during infection with RNA-sequencing (RNA-seq) data or through structural similarity to known effectors are recommended practices for effector candidate prioritization.

Sperschneider, J., Gardiner, D.M., Dodds, P.N., Tini, F., Covarelli, L., Singh, K.B., Manners, J.M. and Taylor, J.M. (2016), EffectorP: predicting fungal effector proteins from secretomes using machine learning. New Phytol, 210: 743-761. https://doi.org/10.1111/nph.13794

##################
 SignalP analysis
##################

setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/signalp-6.0/signalp6_fast/")
Sys.setenv(PATH = paste("/home/barry/.local/bin", Sys.getenv("PATH"), sep = ":"))

SIGNALP_DIR <- system("python3 -c 'import signalp; import os; print(os.path.dirname(signalp.__file__))'", intern = TRUE)

Sys.getenv("PATH")

system("signalp6 -h")

system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")

system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")
Predicting: 100%|██████████| 21334/21334 [39:18<00:00,  9.05sequences/s]

# run on signalP_6.0 and import results 

```{r}
# Read the prediction_results.txt file into a data frame
prediction_results <- read.table("prediction_results.txt", header = F, comment.char = "#", sep = "\t")
colnames(prediction_results) <- c("protein_id", "Prediction", "OTHER", "SP(Sec/SPI)", "CS Position")
head(prediction_results)

# Filter the prediction_results data frame by "SP" in the Prediction column
signalPsecreted <- filter(prediction_results, Prediction == "SP")

# Use separate to split CS Position column
split_results <- signalPsecreted %>%
  separate(`CS Position`, into = c("cleavage_site", "probability"),
           sep = "\\. ", remove = FALSE)

signalPSP <- split_results %>%
  mutate(cleavage_site = gsub("CS pos: ", "", cleavage_site),
         probability = gsub("Pr: ", "", probability))

head(signalPSP)

split_results$probability <- as.numeric(signalPSP$probability)
hist(split_results$probability, main = "Signal peptide probability", xlab = "Probability")

#filter out low probability SP
signalPSP.filtered <- subset(signalPSP, probability >= 0.7)

signalPSP.filtered.id <- dplyr::right_join(gtf.faa.omicsbox, signalPSP.filtered, by = 'protein_id')
#1878/21334 proteins predicted to contain a Sec/SPI SPs using signalP6.0 
#1694/1878 with a probability of 0.7 or greater
```

```{r}
# Get unique protein IDs in phobius.id and signalPSP.filtered.id
uniquephobius <- setdiff(phobius.id$protein_id, signalPSP.filtered.id$protein_id)
uniquephobius.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% uniquephobius, ]
#1361 proteins with signal peptides that are not Sec/SPI SPs using phobius

signalpeptides <- unique(c(phobius.id$protein_id, signalPSP.filtered$protein_id))
signalpeptides.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% signalpeptides, ]
#3055 peptides have a predicted signal peptide
```

```{r}
#list of proteins from interproscan - phobius/tmhmm and signalP predicted to be secreted
phobius.tmhmm.signalp <- unique(c(phobius.id$protein_id, filtered_interproscanTMHMM$protein_id, signalPSP.filtered$protein_id))

pts.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% phobius.tmhmm.signalp, ]
#3373 predicted secreted proteins using EffectorP3.0 suggestion
```

################
 bdCAN analysis
################

https://bcb.unl.edu/dbCAN2/blast.php using allpcgenes.fasta

check all boxes except substrate prediction
HMMER: dbCAN (E-Value < 1e-15, coverage > 0.35)
DIAMOND: CAZy (E-Value < 1e-102
HMMER: dbCAN-sub (E-Value < 1e-15, coverage > 0.35)

Jinfang Zheng and others, dbCAN3: automated carbohydrate-active enzyme and substrate annotation, 
Nucleic Acids Research, 2023;, gkad328, https://doi.org/10.1093/nar/gkad328

```{r}
dbcanresults <- read_tsv("dbCAN_allpcgenes.txt", show_col_types = FALSE) 

colnames(dbcanresults)[colnames(dbcanresults) == "Gene ID"] <- "protein_id"

dbcan <- dbcanresults %>%
  filter(`#ofTools` >= 2)

dbcan.id <- dplyr::right_join(gtf.faa.omicsbox, dbcan, by = 'protein_id')

#which CAZymes do not have signal peptides
dbcansecreted <- setdiff(dbcan.id$protein_id, signalpeptides.id$protein_id)
dbcansecreted.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% dbcansecreted, ]
#435 predicted CAZymes version dbCAN3.0
#paper predicted 468 they had HMMER/Hotpep/DIAMOND  Hotpep replaced with eCAMI obsolete version dbCAN2.0
#179 predicted CAZymes have signal peptides
```

```{r}
totalsecreted <- unique(c(pts.id$protein_id, pfameffectors.id$protein_id, knowneffectors.id$protein_id, signalPSP.filtered.id$protein_id, dbcan.id$protein_id, phobius.id$protein_id, filtered_interproscanTMHMM.id$protein_id))

totalsecreted.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% totalsecreted, ]
```

```{r}
source("../shared_scripts/detected_pcgenes.R", 
       local = knitr::knit_global())

detectedgenes <- as.data.frame(rownames(assay(dds)))
colnames(detectedgenes) <- "gene_id"
detectedgenes.id <- dplyr::right_join(gtf.faa.omicsbox, detectedgenes, by = 'gene_id')

inplantadetected <- unique(detectedgenes.id$protein_id)
inplantasecreted.id <- totalsecreted.id[totalsecreted.id$protein_id %in% inplantadetected, ]
```

```{r, secretome eulerr plot}

s1 <- unique(pfameffectors.id$protein_id)
s2 <- unique(knowneffectors.id$protein_id)
s3 <- unique(signalPSP.filtered.id$protein_id)
s4 <- unique(dbcan.id$protein_id)
s6 <- unique(phobius.id$protein_id)
s7 <- unique(filtered_interproscanTMHMM.id$protein_id)

filename <- "predictedsecretome.png"
png(filename)

width <- 10  
height <- 6  

png(filename, width = width, height = height, units = "in", res = 300)

SECRETOME <- list("Pfam"=s1, "Genome"=s2, "SignalP"=s3, "CAZyme"=s4,"Phobius"=s6, "TMHMM"=s7)

plot(euler(SECRETOME),quantities = TRUE, labels = list(font = 24), main="Predicted secretome")

dev.off()
```

#####################
 EffectorP analysis 
#####################

```{r}
output_file <- "../PcEffector//EffectorP_3.0.0-beta/totalsecreted.fasta"

generate_fasta_file(totalsecreted.id, output_file)

#path to where the 
setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/EffectorP_3.0.0-beta/")

system("/home/barry/.virtualenvs/my-python/bin/python EffectorP.py -i totalsecreted.fasta > totalsecreted_effectorPresults.txt")
```

3615 proteins were provided as input in the following file: totalsecreted.fasta
-----------------
Number of predicted effectors: 1315
Number of predicted cytoplasmic effectors: 882
Number of predicted apoplastic effectors: 433
-----------------
36.4 percent are predicted effectors.
24.4 percent are predicted cytoplasmic effectors.
12.0 percent are predicted apoplastic effectors.
-----------------
4.2 percent of the cytoplasmic effectors are predicted dual-localized (cytoplasmic/apoplastic).
10.9 percent of the apoplastic effectors are predicted dual-localized (apoplastic/cytoplasmic).

```{r}
effectorptotal.results <- read.delim("EffectorP_3.0.0-beta/totalsecreted_effectorPresults.txt", sep = "\t", stringsAsFactors = FALSE, skip = 10)
effectorptotal.results <- effectorptotal.results[1:(nrow(effectorptotal.results) - 15), ]
colnames(effectorptotal.results) <- c("protein_id", "Cytoplasmic_effector", "Apoplastic_effector", "Non_effector", "Prediction")
effectorptotal.results$protein_id <- gsub(" ", "", effectorptotal.results$protein_id)

head(effectorptotal.results)

# make this all non effectors from effectorP
predictedtotaleffectors <- subset(effectorptotal.results, Prediction != "Non-effector")
predictedtotaleffectors.id <- dplyr::right_join(gtf.faa.omicsbox, predictedtotaleffectors, by = 'protein_id')

```

```{r}
output_file <- "../PcEffector//EffectorP_3.0.0-beta/inplantasecreted.fasta"

generate_fasta_file(inplantasecreted.id, output_file)

#path to where the 
setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/EffectorP_3.0.0-beta/")

system("/home/barry/.virtualenvs/my-python/bin/python EffectorP.py -i inplantasecreted.fasta > inplantasecreted_effectorPresults.txt")

```
2607 proteins were provided as input in the following file: inplantasecreted.fasta
-----------------
Number of predicted effectors: 824
Number of predicted cytoplasmic effectors: 582
Number of predicted apoplastic effectors: 242
-----------------
31.6 percent are predicted effectors.
22.3 percent are predicted cytoplasmic effectors.
9.3 percent are predicted apoplastic effectors.
-----------------
3.6 percent of the cytoplasmic effectors are predicted dual-localized (cytoplasmic/apoplastic).
12.0 percent of the apoplastic effectors are predicted dual-localized (apoplastic/cytoplasmic).

```{r}
effectorpinplanta.results <- read.delim("EffectorP_3.0.0-beta/inplantasecreted_effectorPresults.txt", sep = "\t", stringsAsFactors = FALSE, skip = 10)
effectorpinplanta.results <- effectorpinplanta.results[1:(nrow(effectorpinplanta.results) - 15), ]
colnames(effectorpinplanta.results) <- c("protein_id", "Cytoplasmic_effector", "Apoplastic_effector", "Non_effector", "Prediction")
effectorpinplanta.results$protein_id <- gsub(" ", "", effectorpinplanta.results$protein_id)

head(effectorpinplanta.results)
inplantasecreted <- effectorpinplanta.results
inplantasecreted.id <- dplyr::right_join(gtf.faa.omicsbox, inplantasecreted, by = 'protein_id')
# 2607 secreted proteins in planta

inplantanoneffector <- subset(inplantasecreted, inplantasecreted$Prediction == "Non-effector")
inplantanoneffector.id <- dplyr::right_join(gtf.faa.omicsbox, inplantanoneffector, by = 'protein_id')
# 1783 predicted non-effectors

inplantaeffectors <- subset(inplantasecreted, inplantasecreted$Prediction != "Non-effector")
inplantaeffectors.id <- dplyr::right_join(gtf.faa.omicsbox, inplantaeffectors, by = 'protein_id')
# 824 effector genes in planta
```

```{r, effector expression}
#load("Pcdeseq2.RData")

earlydetected <- as.data.frame(res_early_vs_vegetative)
earlydetected$gene_id <- rownames(earlydetected)
earlysecreted <- dplyr::right_join(earlydetected, inplantasecreted.id, by = 'gene_id')

middetected <- as.data.frame(res_middle_vs_vegetative)
middetected$gene_id <- rownames(middetected)
midsecreted <- dplyr::right_join(middetected, inplantasecreted.id, by = 'gene_id')

latedetected <- as.data.frame(res_late_vs_vegetative)
latedetected$gene_id <- rownames(latedetected)
latesecreted <- dplyr::right_join(latedetected, inplantasecreted.id, by = 'gene_id')

```

```{r, effector eulerr plot}

eeup <- earlysecreted[earlysecreted$log2FoldChange > 0 & earlysecreted$padj < 0.05, ]
eeup <- unique(eeup$gene_id)
eedn <- earlysecreted[earlysecreted$log2FoldChange < 0 & earlysecreted$padj < 0.05, ]
eedn <- unique(eedn$gene_id)

meup <- midsecreted[midsecreted$log2FoldChange > 0 & midsecreted$padj < 0.05, ]
meup <- unique(meup$gene_id)
medn <- midsecreted[midsecreted$log2FoldChange < 0 & midsecreted$padj < 0.05, ]
medn <- unique(medn$gene_id)

leup <- latesecreted[latesecreted$log2FoldChange > 0 & latesecreted$padj < 0.05, ]
leup <- unique(leup$gene_id)
ledn <- latesecreted[latesecreted$log2FoldChange < 0 & latesecreted$padj < 0.05, ]
ledn <- unique(ledn$gene_id)

filename <- "Effectorexpression.png"

# Create a PNG device
png(filename)

width <- 10  
height <- 6  

png(filename, width = width, height = height, units = "in", res = 300)

secreted <- list("Earlyup"=eeup, "Earlydn"=eedn, "Midup"=meup, "Middn"=medn, "Lateup"=leup,"Latedn"=ledn)

plot(euler(secreted),quantities = TRUE, labels = list(font = 24), main="In Planta secretome expression")

dev.off()
```

```{r, effector ORA}

bg <- detectedgenes.id$gene_id
bg.set <- data.frame("background", bg)
colnames(bg.set) <- c("term", "gene")

all.set <- data.frame("Secreted", inplantasecreted.id$gene_id)
colnames(all.set) <- c("term", "gene")

Noneffector.set <- data.frame("Non_effector", inplantanoneffector.id$gene_id)
colnames(Noneffector.set) <- c("term", "gene")

apop <- inplantasecreted.id[grep("Y", inplantasecreted.id$Apoplastic_effector),]
apop.set <- data.frame("Apoplastic", apop$gene_id)
colnames(apop.set) <- c("term", "gene")

cyto <- inplantasecreted.id[grep("Y", inplantasecreted.id$Cytoplasmic_effector),]
cyto.set <- data.frame("Cytosolic", cyto$gene_id) 
colnames(cyto.set) <- c("term", "gene")

sets <- rbind(bg.set, all.set, Noneffector.set, apop.set, cyto.set)

```

```{r}
library(clusterProfiler)

ora_res <- as.data.frame(enricher(gene = leup,
                                    universe = bg,
                                    maxGSSize = 50000,
                                    TERM2GENE = sets,
                                    pAdjustMethod = "fdr",
                                    pvalueCutoff = 1,
                                    qvalueCutoff = 1))

ora_res$geneID <- NULL
  ora_res_names <- rownames(ora_res)
  
  ora_res$GeneRatio <- as.character(ora_res$GeneRatio)
  
  gr <- as.numeric(sapply(strsplit(as.character(ora_res$GeneRatio), "/"), "[[", 1)) /
    as.numeric(sapply(strsplit(as.character(ora_res$GeneRatio), "/"), "[[", 2))
  
  br <- as.numeric(sapply(strsplit(as.character(ora_res$BgRatio), "/"), "[[", 1)) /
    as.numeric(sapply(strsplit(as.character(ora_res$BgRatio), "/"), "[[", 2))
  
  ora_res$es <- gr/br
  ora_res <- ora_res[order(-ora_res$es), ]
  ora_res$Description <- NULL
  
view(ora_res)  

leESraw <- ora_res
```

```{r}
# Create a data frame for eeES, meES, and leES
eeES <- data.frame(
  Dataset = "eeES",
  GeneSet = c("Cytosolic", "Apoplastic", "Non_effector"),
  ES = c(7.335492, 6.962049, 4.525169)
)

meES <- data.frame(
  Dataset = "meES",
  GeneSet = c("Cytosolic", "Apoplastic", "Non_effector"),
  ES = c(6.932619, 6.749850, 4.776168)
)

leES <- data.frame(
  Dataset = "leES",
  GeneSet = c("Cytosolic", "Apoplastic", "Non_effector"),
  ES = c(6.621752, 6.713086, 4.919242)
)

# Combine the datasets
combined_data <- rbind(eeES, meES, leES)

# Define the order and labels for the x-axis
dataset_order <- c("eeES", "meES", "leES")
x_labels <- c("Early", "Middle", "Late")

# Convert the "Dataset" column to a factor with the desired order and labels
combined_data$Dataset <- factor(combined_data$Dataset, levels = dataset_order, labels = x_labels)

# Set the filename for the PNG file
filename <- "cytoplasmic-apoplastic_effector1.png"

# Create a PNG device
png(filename)

# Create the line plot
ggplot(combined_data, aes(x = Dataset, y = ES, color = GeneSet, group = GeneSet)) +
  geom_line() +
  geom_point() +
  labs(x = "Time Point", y = "Enrichment Score (ES)",
       title = "Over-representation analysis of predicted secreted genes") +
  scale_color_manual(values = c("Cytosolic" = "blue", "Apoplastic" = "red", "Non_effector" = "black")) +
  theme_minimal()

dev.off()
```
```{r}
genesets <- read.gmt("goterms.gmt")
bg <- detectedgenes.id$gene_id

ora_res <- as.data.frame(enricher(gene = eeup,
                                    universe = bg,
                                    maxGSSize = 50000,
                                    TERM2GENE = genesets,
                                    pAdjustMethod = "fdr",
                                    pvalueCutoff = 1,
                                    qvalueCutoff = 1))

ora_res$geneID <- NULL
  ora_res_names <- rownames(ora_res)
  
  ora_res$GeneRatio <- as.character(ora_res$GeneRatio)
  
  gr <- as.numeric(sapply(strsplit(as.character(ora_res$GeneRatio), "/"), "[[", 1)) /
    as.numeric(sapply(strsplit(as.character(ora_res$GeneRatio), "/"), "[[", 2))
  
  br <- as.numeric(sapply(strsplit(as.character(ora_res$BgRatio), "/"), "[[", 1)) /
    as.numeric(sapply(strsplit(as.character(ora_res$BgRatio), "/"), "[[", 2))
  
  ora_res$es <- gr/br
  ora_res <- ora_res[order(-ora_res$es), ]
  ora_res$Description <- NULL
  
view(ora_res)  

eeESgo <- ora_res
```

```{r}
library(ggplot2)

esig <- subset(earlysecreted, padj < 0.05)
enonsig <- subset(earlysecreted, padj > 0.05)
msig <- subset(midsecreted, padj < 0.05)
mnonsig <- subset(midsecreted, padj > 0.05)
lsig <- subset(latesecreted, padj < 0.05)
lnonsig <- subset(latesecreted, padj > 0.05)


plot(log10(res_early_vs_vegetative$baseMean), res_early_vs_vegetative$log2FoldChange,
     xlab = "log10 basemean", ylab = "log2 foldchange",
     pch = 19, cex = 0.5, col = "dark gray",
     main = "Early effectors")
points(log10(esig$baseMean), esig$log2FoldChange,
       pch = 19, cex = 0.5, col = "blue")
points(log10(enonsig$baseMean), enonsig$log2FoldChange,
       pch = 19, cex = 0.5, col = "red")


plot(log10(res_middle_vs_vegetative$baseMean), res_middle_vs_vegetative$log2FoldChange,
     xlab = "log10 basemean", ylab = "log2 foldchange",
     pch = 19, cex = 0.5, col = "dark gray",
     main = "Mid effectors")
points(log10(msig$baseMean), msig$log2FoldChange,
       pch = 19, cex = 0.5, col = "blue")
points(log10(mnonsig$baseMean), mnonsig$log2FoldChange,
       pch = 19, cex = 0.5, col = "red")


plot(log10(res_late_vs_vegetative$baseMean), res_late_vs_vegetative$log2FoldChange,
     xlab = "log10 basemean", ylab = "log2 foldchange",
     pch = 19, cex = 0.5, col = "dark gray",
     main = "Late effectors")
points(log10(lsig$baseMean), lsig$log2FoldChange,
       pch = 19, cex = 0.5, col = "blue")
points(log10(lnonsig$baseMean), lnonsig$log2FoldChange,
       pch = 19, cex = 0.5, col = "red")

```

```{r}
library(ggplot2)

# Extract columns from DESeqResults object
res_df <- data.frame(
  baseMean = res_early_vs_vegetative$baseMean,
  log2FoldChange = res_early_vs_vegetative$log2FoldChange
)

# Create the smear plot
ggplot(res_df, aes(x = log10(baseMean), y = log2FoldChange)) +
  geom_point(color = "darkgray", alpha = 0.5) +
  xlab("log10 basemean") +
  ylab("log2 foldchange") +
  ggtitle("Smear Plot") +
  
  # Add significant points in blue
  geom_point(data = esig, aes(x = log10(baseMean), y = log2FoldChange),
             color = "blue", alpha = 0.5) +
  
  # Add non-significant points in red
  geom_point(data = enonsig, aes(x = log10(baseMean), y = log2FoldChange),
             color = "red", alpha = 0.5)

```


smear plot that shows the effectors. x=log10 base mean, y=log2 fold change. It will indicate which effectors have the largest overall expression and fold change. You can label the significant effectors red, the non significant effectors blue and all other genes grey in the background.  For the significant effectors with the largest basemean expression, you can add a gene label. You can focus on those ones in your discussion perhaps.

```{r}
library(plotly)

# Convert DESeqResults object to a data frame
res_df <- as.data.frame(res_early_vs_vegetative)

# Create the interactive smear plot
plot_ly(res_df, x = ~log10(baseMean), y = ~log2FoldChange, color = I("darkgray"),
        type = "scatter", mode = "markers", marker = list(opacity = 0.5)) %>%
  add_markers(data = esig, color = I("blue"), hovertext = ~gene_id, hoverinfo = "text") %>%
  add_markers(data = enonsig, color = I("red"), hovertext = ~gene_id, hoverinfo = "text") %>%
  layout(xaxis = list(title = "log10 basemean"),
         yaxis = list(title = "log2 foldchange"),
         title = "Smear Plot")
```



## Session information

```{r,session}

sessionInfo()

```
