---
title: "Chapter 3:Effector analysis"
author: "BWSchroeter"
date: "2023-06-23"
output:
  word_document: default
  html_document: default
---
 
```{r, libs, message=FALSE, warning=FALSE}

library(tidyverse)
library(gridExtra)
library(reshape2)
library(dplyr)
library(edgeR)
library(data.table)
library(clusterProfiler)
library(ggplot2)
library("eulerr")
library(pheatmap)
```

```{r, include=FALSE}
#import raw files
source("../shared_scripts/pcinnamomi_annotations.R", 
       local = knitr::knit_global())
source("../shared_scripts/sample_names.R", 
       local = knitr::knit_global())
source("../shared_scripts/countmatrix.R", 
       local = knitr::knit_global())

```

```{r}
#function used to make .fasta files for analysis
write_fasta <- function(df, file_path) {
  fasta_string <- paste0(">", df$protein_id, "\n", df$Sequence, "\n")
  writeLines(fasta_string, con = file_path)
}

generate_fasta_file <- function(df, output_file) {
  write_fasta(df, output_file)
}

#all Pc proteins from .faa file -- 21334 genes 
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta"

generate_fasta_file(df, output_file)

```

#InterProScan analysis

barry@ziemann-01:/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan/interproscan-5.62-94.0$ ./interproscan.sh -appl CDD,PANTHER,Pfam,PROSITEPATTERNS,PROSITEPROFILES,SFLD,SMART,TIGRFAM,Phobius,TMHMM,SUPERFAMILY -i /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta -iprlookup --goterms -b /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan_output/
05/06/2023 12:56:46:576 Welcome to InterProScan-5.62-94.0
05/06/2023 12:56:46:577 Running InterProScan v5 in STANDALONE mode... on Linux
05/06/2023 12:56:51:785 RunID: ziemann-01_20230605_125651556_b2fz
05/06/2023 12:57:02:847 Loading file /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta
05/06/2023 12:57:02:848 Running the following analyses:
[CDD-3.20,NCBIfam-11.0,PANTHER-17.0,Pfam-35.0,Phobius-1.01,ProSitePatterns-2022_05,ProSiteProfiles-2022_05,SFLD-4,SMART-9.0,SUPERFAMILY-1.75,TMHMM-2.0c]
Available matches will be retrieved from the pre-calculated match lookup service.

Matches for any sequences that are not represented in the lookup service will be calculated locally.
05/06/2023 12:57:07:851 Uploaded 20850 unique sequences for analysis
05/06/2023 12:59:50:492 87% completed
05/06/2023 13:59:50:678 19% completed
 05/06/2023 14:59:50:824 24% completed
05/06/2023 15:59:51:007 77% completed
05/06/2023 16:12:19:464 100% done:  InterProScan analyses completed 

```{r, message=FALSE, warning=FALSE}
# RxLR, CRN and NLP predicted from Engelbrecht, Juanita, et al. "Genome of the destructive oomycete Phytophthora cinnamomi provides insights into its pathogenicity and adaptive potential." BMC genomics 22.1 (2021): 1-15.

source("../shared_scripts/effectors_from_literature.R", 
       local = knitr::knit_global())

```

```{r}
#extract signal peptides from interproscan results 
interproscanSP <- interproscanresults[interproscanresults$signature_id == "SIGNAL_PEPTIDE",] 
interproscanSP <- unique(interproscanSP$protein_id)

phobius.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% interproscanSP, ]
#Phobius predicts 3050 proteins with predicted signal peptides from 16957 proteins scanned - less than 80 amino acids not scanned by interProScan introduces ribosomal proteins - look at removing them based on er retention signals? further work. Although may be a nuclear signal that is seen in plants that target from chloroplast to nucleus

#3050 proteins
```

```{r}
interproscanTMHMM <- unique(interproscanresults[interproscanresults$signature_id == "TMhelix",])

filtered_interproscanTMHMM <- interproscanTMHMM %>%
  group_by(protein_id) %>%
  filter(n() <= 1) %>% 
  filter(end <= 60)

filtered_interproscanTMHMM.id <- dplyr::right_join(gtf.faa.omicsbox, filtered_interproscanTMHMM, by = 'protein_id')

#TMHMM predicts 594 proteins that aren't considered transmembrane
```

Tmhmm v.2.0 was used to predict transmembrane proteins from the combined predicted secretome (Krogh et al., 2001). If a protein contains at least one transmembrane domain after the first 60 amino acids or at least two transmembrane domains in total, it was labelled as a putative transmembrane protein.

Ranking effector candidates according to their expression during infection with RNA-sequencing (RNA-seq) data or through structural similarity to known effectors are recommended practices for effector candidate prioritization.

Sperschneider, J., Gardiner, D.M., Dodds, P.N., Tini, F., Covarelli, L., Singh, K.B., Manners, J.M. and Taylor, J.M. (2016), EffectorP: predicting fungal effector proteins from secretomes using machine learning. New Phytol, 210: 743-761. https://doi.org/10.1111/nph.13794

#SignalP analysis

setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/signalp-6.0/signalp6_fast/")
Sys.setenv(PATH = paste("/home/barry/.local/bin", Sys.getenv("PATH"), sep = ":"))

SIGNALP_DIR <- system("python3 -c 'import signalp; import os; print(os.path.dirname(signalp.__file__))'", intern = TRUE)

Sys.getenv("PATH")
system("signalp6 -h")

system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")

system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")
Predicting: 100%|██████████| 21334/21334 [39:18<00:00,  9.05sequences/s]

run on signalP_6.0 and import results 

```{r}
# Read the prediction_results.txt file into a data frame
prediction_results <- read.table("prediction_results.txt", header = F, comment.char = "#", sep = "\t")
colnames(prediction_results) <- c("protein_id", "Prediction", "OTHER", "SP(Sec/SPI)", "CS Position")
head(prediction_results)

# Filter the prediction_results data frame by "SP" in the Prediction column
signalPsecreted <- filter(prediction_results, Prediction == "SP")

# Use separate to split CS Position column
split_results <- signalPsecreted %>%
  separate(`CS Position`, into = c("cleavage_site", "probability"),
           sep = "\\. ", remove = FALSE)

signalPSP <- split_results %>%
  mutate(cleavage_site = gsub("CS pos: ", "", cleavage_site),
         probability = gsub("Pr: ", "", probability))

head(signalPSP)

split_results$probability <- as.numeric(signalPSP$probability)
hist(split_results$probability, main = "Signal peptide probability", xlab = "Probability")

#filter out low probability SP
signalPSP.filtered <- subset(signalPSP, probability >= 0.7)

signalPSP.filtered.id <- dplyr::right_join(gtf.faa.omicsbox, signalPSP.filtered, by = 'protein_id')

usignalPSP.filtered.id <- unique(signalPSP.filtered.id$protein_id)
#1878/21334 proteins predicted to contain a Sec/SPI SPs using signalP6.0 
#1694/1878 with a probability of 0.7 or greater
```

```{r}
# Get unique protein IDs in phobius.id and signalPSP.filtered.id
uniquephobius <- setdiff(phobius.id$protein_id, signalPSP.filtered.id$protein_id)
uniquephobius.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% uniquephobius, ]
#1361 proteins with signal peptides that are not Sec/SPI SPs using phobius

signalpeptides <- unique(c(phobius.id$protein_id, signalPSP.filtered$protein_id))

signalpeptides.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% signalpeptides, ]
#3055 peptides have a predicted signal peptide
```

```{r}
#list of proteins from interproscan - phobius/tmhmm and signalP predicted to be secreted
phobius.tmhmm.signalp <- unique(c(phobius.id$protein_id, filtered_interproscanTMHMM$protein_id, signalPSP.filtered$protein_id))

pts.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% phobius.tmhmm.signalp, ]
#3373 predicted secreted proteins using EffectorP3.0 suggestion
```

#dbCAN analysis

https://bcb.unl.edu/dbCAN2/blast.php using allpcgenes.fasta

check all boxes except substrate prediction
HMMER: dbCAN (E-Value < 1e-15, coverage > 0.35)
DIAMOND: CAZy (E-Value < 1e-102
HMMER: dbCAN-sub (E-Value < 1e-15, coverage > 0.35)

Jinfang Zheng and others, dbCAN3: automated carbohydrate-active enzyme and substrate annotation, 
Nucleic Acids Research, 2023;, gkad328, https://doi.org/10.1093/nar/gkad328

```{r}
dbcanresults <- read_tsv("dbCAN_allpcgenes.txt", show_col_types = FALSE) 

colnames(dbcanresults)[colnames(dbcanresults) == "Gene ID"] <- "protein_id"

dbcan <- dbcanresults %>%
  mutate(CAZyme = ifelse(`#ofTools` >= 2, "Yes", "No"))

dbcan.filter <- dbcan %>%
  filter(`#ofTools` >= 2) 

dbcan.id <- dplyr::right_join(gtf.faa.omicsbox, dbcan.filter, by = 'protein_id')

#which CAZymes do not have signal peptides
dbcansecreted <- setdiff(dbcan.id$protein_id, signalpeptides.id$protein_id)
dbcansecreted.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% dbcansecreted, ]
#435 predicted CAZymes version dbCAN3.0
#paper predicted 468 they had HMMER/Hotpep/DIAMOND  Hotpep replaced with eCAMI obsolete version dbCAN2.0
#179 predicted CAZymes have signal peptides
```

```{r}
totalsecreted <- unique(c(pts.id$protein_id, pfameffectors.id$protein_id, knowneffectors.id$protein_id, signalPSP.filtered.id$protein_id, dbcan.id$protein_id, phobius.id$protein_id, filtered_interproscanTMHMM.id$protein_id))

totalsecreted.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% totalsecreted, ]

```

```{r, warning=FALSE}
source("../shared_scripts/detected_pcgenes.R", 
       local = knitr::knit_global())

detectedgenes <- as.data.frame(rownames(assay(dds)))
colnames(detectedgenes) <- "gene_id"

inplantadetected <- unique(detectedgenes.id$protein_id)
inplantasecreted.id <- totalsecreted.id[totalsecreted.id$protein_id %in% inplantadetected, ]
```

```{r, secretome eulerr plot, warning=FALSE}

s1 <- unique(pfameffectors.id$protein_id)
s2 <- unique(knowneffectors.id$protein_id)
s3 <- unique(signalPSP.filtered.id$protein_id)
s4 <- unique(dbcan.id$protein_id)
s6 <- unique(phobius.id$protein_id)
s7 <- unique(filtered_interproscanTMHMM.id$protein_id)

png("../Figures/predictedsecretome.png", width = 10, height = 6, units = "in", res = 600)

SECRETOME <- list("Pfam"=s1, "Genome"=s2, "SignalP"=s3, "CAZyme"=s4,"Phobius"=s6, "TMHMM"=s7)

plot(euler(SECRETOME),quantities = TRUE, labels = list(font = 24), main="")

dev.off()
```

```{r}
signalPSP_protein_ids <- signalPSP.filtered.id$protein_id
phobius_protein_ids <- phobius.id$protein_id

# Find the protein IDs that are unique to the phobius.id dataframe
unique_phobius_ids <- subset(phobius.id, !(protein_id %in% signalPSP_protein_ids))
```

#EffectorP analysis 

```{r, warning=FALSE}
output_file <- "../PcEffector//EffectorP_3.0.0-beta/inplantasecreted.fasta"

generate_fasta_file(inplantasecreted.id, output_file)

setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/EffectorP_3.0.0-beta/")

system("/home/barry/.virtualenvs/my-python/bin/python EffectorP.py -i inplantasecreted.fasta > inplantasecreted_effectorPresults.txt")

```

```{r}
effectorpinplanta.output <- tail(read.delim("EffectorP_3.0.0-beta/inplantasecreted_effectorPresults.txt", sep = "\t", stringsAsFactors = FALSE), 15)
view(effectorpinplanta.output)
```

```{r,echo=FALSE}
effectorpinplanta.results <- read.delim("EffectorP_3.0.0-beta/inplantasecreted_effectorPresults.txt", sep = "\t", stringsAsFactors = FALSE, skip = 10)
effectorpinplanta.results <- effectorpinplanta.results[1:(nrow(effectorpinplanta.results) - 15), ]
colnames(effectorpinplanta.results) <- c("protein_id", "Cytoplasmic_effector", "Apoplastic_effector", "Non_effector", "Prediction")
effectorpinplanta.results$protein_id <- gsub(" ", "", effectorpinplanta.results$protein_id)

head(effectorpinplanta.results)
inplantasecreted <- effectorpinplanta.results
inplantasecreted.id <- dplyr::right_join(gtf.faa.omicsbox, inplantasecreted, by = 'protein_id')
# 2607 secreted proteins in planta

inplantanoneffector <- subset(inplantasecreted, inplantasecreted$Prediction == "Non-effector")
inplantanoneffector.id <- dplyr::right_join(gtf.faa.omicsbox, inplantanoneffector, by = 'protein_id')
# 1783 predicted non-effectors

inplantaeffectors <- subset(inplantasecreted, inplantasecreted$Prediction != "Non-effector")
inplantaeffectors.id <- dplyr::right_join(gtf.faa.omicsbox, inplantaeffectors, by = 'protein_id')
# 824 effector genes in planta
```

```{r}
# create a dataframe that has secreted, number of tmhelices and signal peptide

signalpeptides.id <- signalpeptides.id %>%
  mutate(SP = "Yes")

gtf.omicsbox.effector <-  gtf.faa.omicsbox %>%
  left_join(signalpeptides.id %>%
              select(protein_id, SP), by = "protein_id") %>%
  mutate(SP = ifelse(is.na(SP), "No", SP))

TMHMMcount <- interproscanresults[interproscanresults$analysis == "TMHMM",]

TMHMMcount <- TMHMMcount %>%
  group_by(protein_id) %>%
  mutate(tmhelices = n())

TMHMMcount <- TMHMMcount %>%
  distinct(protein_id, .keep_all = TRUE)

tmhelices_data <- TMHMMcount %>%
  select(protein_id, tmhelices)

gtf.omicsbox.effector <- gtf.omicsbox.effector %>%
  left_join(tmhelices_data, by = "protein_id") 

gtf.omicsbox.effector$tmhelices <- ifelse(is.na(gtf.omicsbox.effector$tmhelices), 0, gtf.omicsbox.effector$tmhelices)

totalsecreted.id <- totalsecreted.id %>%
  mutate(secreted = "Yes")

gtf.omicsbox.effector <-  gtf.omicsbox.effector %>%
  left_join(totalsecreted.id %>%
              select(protein_id, secreted), by = "protein_id") %>%
  mutate(secreted = ifelse(is.na(secreted), "No", secreted))

gtf.omicsbox.effector <-  gtf.omicsbox.effector %>%
  left_join(inplantasecreted %>%
              select(protein_id, Prediction), by = "protein_id") 

gtf.omicsbox.effector$Prediction <- ifelse(is.na(gtf.omicsbox.effector$Prediction), "-", gtf.omicsbox.effector$Prediction)

gtf.omicsbox.effector <- gtf.omicsbox.effector %>%
  mutate(`SP, TM` = paste(SP, tmhelices, sep = ", "))

```

```{r}
# add PHI blast results and use for TCseq cluster analysis
phiblast <- read_delim("PHIB_results.tsv")

gtf.omicsbox.effector.phi <- dplyr::right_join(phiblast, gtf.omicsbox.effector, by = 'protein_id')

write.csv(gtf.omicsbox.effector.phi, file = "/mnt/data/barry/projects/Schroeter_Pcinnamomi/Timecourse analysis/effectortable.tsv", row.names = FALSE)

```


```{r, add significance columns}
earlydesig <- as.data.frame(res_early_vs_vegetative)
earlydesig$significant <- ""

middesig <- as.data.frame(res_middle_vs_vegetative)
middesig$significant <- ""

latedesig <- as.data.frame(res_late_vs_vegetative)
latedesig$significant <- ""

# Update 'significant' column based on your conditions
earlydesig$significant[earlydesig$log2FoldChange > 1 & earlydesig$padj < 0.05 | earlydesig$log2FoldChange < -1 & earlydesig$padj < 0.05] <- "●"
middesig$significant[middesig$log2FoldChange > 1 & middesig$padj < 0.05 | middesig$log2FoldChange < -1 & middesig$padj < 0.05] <- "●"
latedesig$significant[latedesig$log2FoldChange > 1 & latedesig$padj < 0.05 | latedesig$log2FoldChange < -1 & latedesig$padj < 0.05] <- "●"
```

```{r, effector expression, echo=FALSE}

earlydetected <- as.data.frame(earlydesig)
earlydetected$gene_id <- rownames(earlydesig)
earlysecreted <- dplyr::right_join(earlydetected, inplantasecreted.id, by = 'gene_id')

middetected <- as.data.frame(middesig)
middetected$gene_id <- rownames(middetected)
midsecreted <- dplyr::right_join(middetected, inplantasecreted.id, by = 'gene_id')

latedetected <- as.data.frame(latedesig)
latedetected$gene_id <- rownames(latedetected)
latesecreted <- dplyr::right_join(latedetected, inplantasecreted.id, by = 'gene_id')

```

```{r, join DE tables into one}
early_DE <- earlydetected %>%
  select(gene_id, padj, log2FoldChange) %>% 
  `colnames<-`(c("gene_id", "earlypadj", "earlyl2fc"))

mid_DE <- middetected %>%
  select(gene_id, padj, log2FoldChange) %>% 
  `colnames<-`(c("gene_id", "midpadj", "midl2fc"))

late_DE <- latedetected %>%
  select(gene_id, padj, log2FoldChange) %>% 
  `colnames<-`(c("gene_id", "latepadj", "latel2fc"))

merged_DE_data <- early_DE %>%
  left_join(mid_DE, by = "gene_id") %>%
  left_join(late_DE, by = "gene_id")

secreted_DE <-dplyr::right_join(merged_DE_data, inplantasecreted.id, by = 'gene_id')

na_count <- sum(is.na(secreted_DE$earlypadj))

# Print the result
print(na_count)
#remove 105 earlypadj NA values (BaseMean values very small < 4.795738) 
secreted_DE$earlypadj <- ifelse(is.na(secreted_DE$earlypadj), 1, secreted_DE$earlypadj)

#use this table for barchart script
write.csv(secreted_DE, file = "/mnt/data/barry/projects/Schroeter_Pcinnamomi/shared_scripts/secretetable.tsv", row.names = FALSE)


```

```{r, secretome expression eulerr plot, echo=FALSE}
# plus minus 0 gives detected genes
eeup <- earlysecreted[earlysecreted$log2FoldChange > 0 & earlysecreted$padj < 0.05, ]
eeup <- unique(eeup$gene_id)
eedn <- earlysecreted[earlysecreted$log2FoldChange < 0 & earlysecreted$padj < 0.05, ]
eedn <- unique(eedn$gene_id)

meup <- midsecreted[midsecreted$log2FoldChange > 0 & midsecreted$padj < 0.05, ]
meup <- unique(meup$gene_id)
medn <- midsecreted[midsecreted$log2FoldChange < -0 & midsecreted$padj < 0.05, ]
medn <- unique(medn$gene_id)

leup <- latesecreted[latesecreted$log2FoldChange > 0 & latesecreted$padj < 0.05, ]
leup <- unique(leup$gene_id)
ledn <- latesecreted[latesecreted$log2FoldChange < -0 & latesecreted$padj < 0.05, ]
ledn <- unique(ledn$gene_id)

png("../Figures/Secretomedetectedvenn.png", width = 6400, height = 4800, res = 600)

secreted <- list("Early (↑)"=eeup, "Early (↓)"=eedn, "Mid (↑)"=meup, 
                 "Mid (↓)"=medn, "Late (↑)"=leup,"Late (↓)"=ledn)

plot(euler(secreted), quantities = TRUE, labels = list(font = 26), main=expression(""))

dev.off()
```

```{r, expression data}
dds_genes <- dds

# Extract the expression values for the top genes
dds_gene_expression <- assay(dds)[rownames(dds_genes), ]

# Create a matrix of log2-transformed expression values
log2_dds_expression <- log2(dds_gene_expression + 1)

# Convert the matrix to a dataframe
dds.df <- as.data.frame(log2_dds_expression)

# Create a new column named "gene_id" and assign the rownames to it
dds.df$gene_id <- rownames(dds.df)

# Reset the rownames of the matrix
rownames(dds.df) <- NULL

```

```{r}
col_ann <- colData[,c(1,3)]
rownames(col_ann) <- col_ann$Sample
col_ann <- data.frame(col_ann)
col_ann$Stage <- as.factor(col_ann$Stage)
col_ann <- col_ann[order(col_ann$Stage),]
col_ann$sample_ID <- NULL

col_ann$Stage <- factor(col_ann$Stage, levels = c("Vegetative", "Early", "Middle", "Late"))

# Sort the `col_ann` dataframe based on the ordered `Stage` column
col_ann <- col_ann[order(col_ann$Stage), ]

```

```{r}
cluster_6_gene_ids <- read.csv("/mnt/data/barry/projects/Schroeter_Pcinnamomi/shared_scripts/cluster6.tsv")
#get gene id that are expressed in cluster 6 
cluster_6_gene_ids <- cluster_6_gene_ids$gene_id
```

```{r, warning=FALSE}
RNAhelicases <- interproscanresults[grepl('PF00270', interproscanresults$signature_id), ]

RNAhelicases.id <- dplyr::right_join(gtf.omicsbox.effector.phi, RNAhelicases, by = 'protein_id')

rnahel <- unique(RNAhelicases.id$gene_id)

rnahelede <- earlydetected[earlydetected$gene_id %in% rnahel, ]
rnahelmde <- middetected[middetected$gene_id %in% rnahel, ]
rnahellde <- latedetected[latedetected$gene_id %in% rnahel, ]

# Mutate a new column in RNAhelicases.id to add a '♦' sign for common rows
rnahelede_cluster6 <- rnahelede %>%
  mutate(common_flag = ifelse(gene_id %in% cluster_6_gene_ids, paste0(gene_id, " \u2666"), gene_id))

rnahel_matrix <- matrix(
  c(rnahelede$log2FoldChange, rnahelmde$log2FoldChange, rnahellde$log2FoldChange),
  nrow = length(rnahelede$log2FoldChange),
  ncol = 3,
  byrow = FALSE
)

rnahel_matrixlabel <- matrix(
  c(rnahelede$significant, rnahelmde$significant, rnahellde$significant),
  nrow = length(rnahelede$significant),
  ncol = 3,
  byrow = FALSE
)

rownames(rnahel_matrix) <- rnahelede_cluster6$common_flag
colnames(rnahel_matrix) <- c("Early", "Middle", "Late")

rownames(rnahel_matrix) <- rnahelede_cluster6$common_flag
colnames(rnahel_matrix) <- c("Early", "Middle", "Late")

png("../Figures/RNAhelicasepheatmap.png", width = 3600, height = 6400, res = 600)
# Create the heatmap with row labels
pheatmap(rnahel_matrix, 
         fontsize_row = 10, fontsize_col = 14, 
         cellheight = 10, cellwidth = 80,
         cluster_rows = T,
         treeheight_row = 0,
         display_numbers = rnahel_matrixlabel,
         cluster_cols = F,
         show_rownames = TRUE,
         labels_row = rnahelede_cluster6$common_flag,
         main = "DEAD/DEAH box helicases",
         border_color = "grey")

```


```{r, warning=FALSE}
rxlr <- pcrxlr$gene_id

rxlrede <- earlydetected[earlydetected$gene_id %in% rxlr, ]
rxlrmde <- middetected[middetected$gene_id %in% rxlr, ]
rxlrlde <- latedetected[latedetected$gene_id %in% rxlr, ]

# Mutate a new column in to add a '♦' sign for cluster 6 genes
rxlrede_cluster6 <- rxlrede %>%
  mutate(common_flag = ifelse(gene_id %in% cluster_6_gene_ids, paste0(gene_id, " \u2666"), gene_id))

rxlr_matrix <- matrix(
  c(rxlrede$log2FoldChange, rxlrmde$log2FoldChange, rxlrlde$log2FoldChange),
  nrow = length(rxlrede$log2FoldChange),
  ncol = 3,
  byrow = FALSE
)

rxlr_matrixlabel <- matrix(
  c(rxlrede$significant, rxlrmde$significant, rxlrlde$significant),
  nrow = length(rxlrede$significant),
  ncol = 3,
  byrow = FALSE
)

rownames(rxlr_matrix) <- rxlrede$gene_id
colnames(rxlr_matrix) <- c("Early", "Middle", "Late")

rownames(rxlr_matrixlabel) <- rxlrede$gene_id
colnames(rxlr_matrixlabel) <- c("Early", "Middle", "Late")

filename <- 
png("../Figures/RxLR.png", width = 3600, height = 6400, res = 600)

# Create the heatmap with row labels
pheatmap(rxlr_matrix,
         fontsize_row = 10, fontsize_col = 14, 
         cellheight = 10, cellwidth = 80,
         cluster_rows = TRUE,
         treeheight_row = 0,
         fontsize_number = 10,
         display_numbers = rxlr_matrixlabel,
         cluster_cols = FALSE,
         show_rownames = TRUE,
         labels_row = rxlrede_cluster6$common_flag,
         main = "RxLR",
         border_color = "grey")
dev.off()

```

```{r, warning=FALSE}
crnlog <- pccrn$gene_id

crnede <- earlydetected[earlydetected$gene_id %in% crnlog, ]
crnmde <- middetected[middetected$gene_id %in% crnlog, ]
crnlde <- latedetected[latedetected$gene_id %in% crnlog, ]

# Mutate a new column in to add a '♦' sign for cluster 6 genes
crnede_cluster6 <- crnede %>%
  mutate(common_flag = ifelse(gene_id %in% cluster_6_gene_ids, paste0(gene_id, " \u2666"), gene_id))

crn_matrix <- matrix(
  c(crnede$log2FoldChange, crnmde$log2FoldChange, crnlde$log2FoldChange),
  nrow = length(crnede$log2FoldChange),
  ncol = 3,
  byrow = FALSE
)

crn_matrixlabel <- matrix(
  c(crnede$significant, crnmde$significant, crnlde$significant),
  nrow = length(crnede$significant),
  ncol = 3,
  byrow = FALSE
)

rownames(crn_matrix) <- crnede$gene_id
colnames(crn_matrix) <- c("Early", "Middle", "Late")

png("../Figures/CRNpheatmap.png", width = 3600, height = 6400, res = 600)

# Create the heatmap with row labels
pheatmap(crn_matrix,
         fontsize_row = 10, fontsize_col = 14, 
         cellheight = 10, cellwidth = 80,
         cluster_rows = TRUE,
         treeheight_row = 0,
         fontsize_number = 10,
         cluster_cols = FALSE,
         display_numbers = crn_matrixlabel,
         show_rownames = TRUE,
         labels_row = crnede_cluster6$common_flag,
         main = "CRN",
         border_color = "grey")
dev.off()
```

```{r, warning=FALSE}
nlplog <- pcnlp$gene_id

nlpede <- earlydetected[earlydetected$gene_id %in% nlplog, ]
nlpmde <- middetected[middetected$gene_id %in% nlplog, ]
nlplde <- latedetected[latedetected$gene_id %in% nlplog, ]

# Mutate a new column in to add a '♦' sign for cluster 6 genes
nlpede_cluster6 <- nlpede %>%
  mutate(common_flag = ifelse(gene_id %in% cluster_6_gene_ids, paste0(gene_id, "  \u2666"), gene_id))

nlp_matrix <- matrix(
  c(nlpede$log2FoldChange, nlpmde$log2FoldChange, nlplde$log2FoldChange),
  nrow = length(nlpede$log2FoldChange),
  ncol = 3,
  byrow = FALSE
)

nlp_matrixlabel <- matrix(
  c(nlpede$significant, nlpmde$significant, nlplde$significant),
  nrow = length(nlpede$significant),
  ncol = 3,
  byrow = FALSE
)
rownames(nlp_matrix) <- nlpede$gene_id
colnames(nlp_matrix) <- c("Early", "Middle", "Late")

png("../Figures/NLPpheatmap.png",  width = 3600, height = 6400, res = 600)

# Create the heatmap with row labels
pheatmap(nlp_matrix,
         fontsize_row = 10, fontsize_col = 14, 
         cellheight = 10, cellwidth = 80,
         cluster_rows = TRUE,
         treeheight_row = 0,
         fontsize_number = 10,
         cluster_cols = FALSE,
         display_numbers = nlp_matrixlabel,
         show_rownames = TRUE,
         labels_row = nlpede_cluster6$common_flag,
         main = "NLP",
         border_color = "grey")
dev.off()
```

```{r, warning=FALSE}
PF00964. <-PF00964$protein_id
PF00964.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% PF00964., ]

pfamlog <- PF00964.id$gene_id 

pfamede <- earlydetected[earlydetected$gene_id %in% pfamlog, ]
pfammde <- middetected[middetected$gene_id %in% pfamlog, ]
pfamlde <- latedetected[latedetected$gene_id %in% pfamlog, ]

# Mutate a new column in to add a '♦' sign for cluster 6 genes
pfamede_cluster6 <- pfamede %>%
  mutate(common_flag = ifelse(gene_id %in% cluster_6_gene_ids, paste0(gene_id, " \u2666"), gene_id))

pfam_matrix <- matrix(
  c(pfamede$log2FoldChange, pfammde$log2FoldChange, pfamlde$log2FoldChange),
  nrow = length(pfamede$log2FoldChange),
  ncol = 3,
  byrow = FALSE
)

pfam_matrixlabel <- matrix(
  c(pfamede$significant, pfammde$significant, pfamlde$significant),
  nrow = length(pfamede$significant),
  ncol = 3,
  byrow = FALSE
)
rownames(pfam_matrix) <- pfamede$gene_id
colnames(pfam_matrix) <- c("Early", "Middle", "Late")

png("../Figures/Elicitinpheatmap.png", width = 3600, height = 6400, res = 600)

# Create the heatmap with row labels
pheatmap(pfam_matrix,
         fontsize_row = 10, fontsize_col = 14, 
         cellheight = 10, cellwidth = 80,
         cluster_rows = TRUE,
         treeheight_row = 0,
         fontsize_number = 10,
         cluster_cols = FALSE,
         display_numbers = pfam_matrixlabel,
         show_rownames = TRUE,
         labels_row = pfamede_cluster6$common_flag,
         main = "Elicitin",
         border_color = "grey")
dev.off()

```

## Session information

```{r,session}

sessionInfo()

```
