---
title: "Chapter 3:Effector analysis"
author: "BWSchroeter"
date: "2023-06-23"
output:
  word_document: default
  html_document: default
---
 
```{r, libs, message=FALSE, warning=FALSE}

library(tidyverse)
library(gridExtra)
library(reshape2)
library(dplyr)
library(edgeR)
library(data.table)
library(clusterProfiler)
library(ggplot2)
#library("kableExtra")
library("eulerr")
library(pheatmap)
```

```{r, include=FALSE}
#import raw files
source("../shared_scripts/pcinnamomi_annotations.R", 
       local = knitr::knit_global())
source("../shared_scripts/sample_names.R", 
       local = knitr::knit_global())
source("../shared_scripts/countmatrix.R", 
       local = knitr::knit_global())

```

```{r}
#function used to make .fasta files for analysis
write_fasta <- function(df, file_path) {
  fasta_string <- paste0(">", df$protein_id, "\n", df$Sequence, "\n")
  writeLines(fasta_string, con = file_path)
}

generate_fasta_file <- function(df, output_file) {
  write_fasta(df, output_file)
}

#all Pc proteins from .faa file -- 21334 genes 
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta"

generate_fasta_file(df, output_file)

```

#InterProScan analysis

barry@ziemann-01:/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan/interproscan-5.62-94.0$ ./interproscan.sh -appl CDD,PANTHER,Pfam,PROSITEPATTERNS,PROSITEPROFILES,SFLD,SMART,TIGRFAM,Phobius,TMHMM,SUPERFAMILY -i /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta -iprlookup --goterms -b /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan_output/
05/06/2023 12:56:46:576 Welcome to InterProScan-5.62-94.0
05/06/2023 12:56:46:577 Running InterProScan v5 in STANDALONE mode... on Linux
05/06/2023 12:56:51:785 RunID: ziemann-01_20230605_125651556_b2fz
05/06/2023 12:57:02:847 Loading file /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta
05/06/2023 12:57:02:848 Running the following analyses:
[CDD-3.20,NCBIfam-11.0,PANTHER-17.0,Pfam-35.0,Phobius-1.01,ProSitePatterns-2022_05,ProSiteProfiles-2022_05,SFLD-4,SMART-9.0,SUPERFAMILY-1.75,TMHMM-2.0c]
Available matches will be retrieved from the pre-calculated match lookup service.

Matches for any sequences that are not represented in the lookup service will be calculated locally.
05/06/2023 12:57:07:851 Uploaded 20850 unique sequences for analysis
05/06/2023 12:59:50:492 87% completed
05/06/2023 13:59:50:678 19% completed
 05/06/2023 14:59:50:824 24% completed
05/06/2023 15:59:51:007 77% completed
05/06/2023 16:12:19:464 100% done:  InterProScan analyses completed 

```{r, message=FALSE, warning=FALSE}
# RxLR, CRN and NLP predicted from Engelbrecht, Juanita, et al. "Genome of the destructive oomycete Phytophthora cinnamomi provides insights into its pathogenicity and adaptive potential." BMC genomics 22.1 (2021): 1-15.

source("../shared_scripts/effectors_from_literature.R", 
       local = knitr::knit_global())

```

```{r}
#extract signal peptides from interproscan results 
interproscanSP <- interproscanresults[interproscanresults$signature_id == "SIGNAL_PEPTIDE",] 
interproscanSP <- unique(interproscanSP$protein_id)

phobius.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% interproscanSP, ]
#Phobius predicts 3050 proteins with predicted signal peptides from 16957 proteins scanned - less than 80 amino acids not scanned by interProScan introduces ribosomal proteins - look at removing them based on er retention signals? further work. Although may be a nuclear signal that is seen in plants that target from chloroplast to nucleus

#3050 proteins
```

```{r}
interproscanTMHMM <- unique(interproscanresults[interproscanresults$signature_id == "TMhelix",])

filtered_interproscanTMHMM <- interproscanTMHMM %>%
  group_by(protein_id) %>%
  filter(n() <= 1) %>% 
  filter(end <= 60)

filtered_interproscanTMHMM.id <- dplyr::right_join(gtf.faa.omicsbox, filtered_interproscanTMHMM, by = 'protein_id')

#TMHMM predicts 594 proteins that aren't considered transmembrane
```

Tmhmm v.2.0 was used to predict transmembrane proteins from the combined predicted secretome (Krogh et al., 2001). If a protein contains at least one transmembrane domain after the first 60 amino acids or at least two transmembrane domains in total, it was labelled as a putative transmembrane protein.

Ranking effector candidates according to their expression during infection with RNA-sequencing (RNA-seq) data or through structural similarity to known effectors are recommended practices for effector candidate prioritization.

Sperschneider, J., Gardiner, D.M., Dodds, P.N., Tini, F., Covarelli, L., Singh, K.B., Manners, J.M. and Taylor, J.M. (2016), EffectorP: predicting fungal effector proteins from secretomes using machine learning. New Phytol, 210: 743-761. https://doi.org/10.1111/nph.13794


#SignalP analysis


setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/signalp-6.0/signalp6_fast/")
Sys.setenv(PATH = paste("/home/barry/.local/bin", Sys.getenv("PATH"), sep = ":"))

SIGNALP_DIR <- system("python3 -c 'import signalp; import os; print(os.path.dirname(signalp.__file__))'", intern = TRUE)

Sys.getenv("PATH")

system("signalp6 -h")

system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")

system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")
Predicting: 100%|██████████| 21334/21334 [39:18<00:00,  9.05sequences/s]

run on signalP_6.0 and import results 

```{r}
# Read the prediction_results.txt file into a data frame
prediction_results <- read.table("prediction_results.txt", header = F, comment.char = "#", sep = "\t")
colnames(prediction_results) <- c("protein_id", "Prediction", "OTHER", "SP(Sec/SPI)", "CS Position")
head(prediction_results)

# Filter the prediction_results data frame by "SP" in the Prediction column
signalPsecreted <- filter(prediction_results, Prediction == "SP")

# Use separate to split CS Position column
split_results <- signalPsecreted %>%
  separate(`CS Position`, into = c("cleavage_site", "probability"),
           sep = "\\. ", remove = FALSE)

signalPSP <- split_results %>%
  mutate(cleavage_site = gsub("CS pos: ", "", cleavage_site),
         probability = gsub("Pr: ", "", probability))

head(signalPSP)

split_results$probability <- as.numeric(signalPSP$probability)
hist(split_results$probability, main = "Signal peptide probability", xlab = "Probability")

#filter out low probability SP
signalPSP.filtered <- subset(signalPSP, probability >= 0.7)

signalPSP.filtered.id <- dplyr::right_join(gtf.faa.omicsbox, signalPSP.filtered, by = 'protein_id')
#1878/21334 proteins predicted to contain a Sec/SPI SPs using signalP6.0 
#1694/1878 with a probability of 0.7 or greater
```

```{r}
# Get unique protein IDs in phobius.id and signalPSP.filtered.id
uniquephobius <- setdiff(phobius.id$protein_id, signalPSP.filtered.id$protein_id)
uniquephobius.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% uniquephobius, ]
#1361 proteins with signal peptides that are not Sec/SPI SPs using phobius

signalpeptides <- unique(c(phobius.id$protein_id, signalPSP.filtered$protein_id))
signalpeptides.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% signalpeptides, ]
#3055 peptides have a predicted signal peptide
```

```{r}
#list of proteins from interproscan - phobius/tmhmm and signalP predicted to be secreted
phobius.tmhmm.signalp <- unique(c(phobius.id$protein_id, filtered_interproscanTMHMM$protein_id, signalPSP.filtered$protein_id))

pts.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% phobius.tmhmm.signalp, ]
#3373 predicted secreted proteins using EffectorP3.0 suggestion
```

#bdCAN analysis

https://bcb.unl.edu/dbCAN2/blast.php using allpcgenes.fasta

check all boxes except substrate prediction
HMMER: dbCAN (E-Value < 1e-15, coverage > 0.35)
DIAMOND: CAZy (E-Value < 1e-102
HMMER: dbCAN-sub (E-Value < 1e-15, coverage > 0.35)

Jinfang Zheng and others, dbCAN3: automated carbohydrate-active enzyme and substrate annotation, 
Nucleic Acids Research, 2023;, gkad328, https://doi.org/10.1093/nar/gkad328

```{r}
dbcanresults <- read_tsv("dbCAN_allpcgenes.txt", show_col_types = FALSE) 

colnames(dbcanresults)[colnames(dbcanresults) == "Gene ID"] <- "protein_id"

dbcan <- dbcanresults %>%
  filter(`#ofTools` >= 2)

dbcan.id <- dplyr::right_join(gtf.faa.omicsbox, dbcan, by = 'protein_id')

#which CAZymes do not have signal peptides
dbcansecreted <- setdiff(dbcan.id$protein_id, signalpeptides.id$protein_id)
dbcansecreted.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% dbcansecreted, ]
#435 predicted CAZymes version dbCAN3.0
#paper predicted 468 they had HMMER/Hotpep/DIAMOND  Hotpep replaced with eCAMI obsolete version dbCAN2.0
#179 predicted CAZymes have signal peptides
```

```{r}
totalsecreted <- unique(c(pts.id$protein_id, pfameffectors.id$protein_id, knowneffectors.id$protein_id, signalPSP.filtered.id$protein_id, dbcan.id$protein_id, phobius.id$protein_id, filtered_interproscanTMHMM.id$protein_id))

totalsecreted.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% totalsecreted, ]
```

```{r, warning=FALSE}
source("../shared_scripts/detected_pcgenes.R", 
       local = knitr::knit_global())

detectedgenes <- as.data.frame(rownames(assay(dds)))
colnames(detectedgenes) <- "gene_id"
detectedgenes.id <- dplyr::right_join(gtf.faa.omicsbox, detectedgenes, by = 'gene_id')

inplantadetected <- unique(detectedgenes.id$protein_id)
inplantasecreted.id <- totalsecreted.id[totalsecreted.id$protein_id %in% inplantadetected, ]
```

```{r, secretome eulerr plot, warning=FALSE}

s1 <- unique(pfameffectors.id$protein_id)
s2 <- unique(knowneffectors.id$protein_id)
s3 <- unique(signalPSP.filtered.id$protein_id)
s4 <- unique(dbcan.id$protein_id)
s6 <- unique(phobius.id$protein_id)
s7 <- unique(filtered_interproscanTMHMM.id$protein_id)

filename <- "predictedsecretome.png"
png(filename)

width <- 10  
height <- 6  

png(filename, width = width, height = height, units = "in", res = 300)

SECRETOME <- list("Pfam"=s1, "Genome"=s2, "SignalP"=s3, "CAZyme"=s4,"Phobius"=s6, "TMHMM"=s7)

plot(euler(SECRETOME),quantities = TRUE, labels = list(font = 24), main="Predicted secretome")

dev.off()
```
```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/predictedsecretome.png")
```

#EffectorP analysis 

```{r, warning=FALSE}
output_file <- "../PcEffector//EffectorP_3.0.0-beta/inplantasecreted.fasta"

generate_fasta_file(inplantasecreted.id, output_file)

setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/EffectorP_3.0.0-beta/")

system("/home/barry/.virtualenvs/my-python/bin/python EffectorP.py -i inplantasecreted.fasta > inplantasecreted_effectorPresults.txt")

```

```{r}
effectorpinplanta.output <- tail(read.delim("EffectorP_3.0.0-beta/inplantasecreted_effectorPresults.txt", sep = "\t", stringsAsFactors = FALSE), 15)
view(effectorpinplanta.output)
```


```{r,echo=FALSE}
effectorpinplanta.results <- read.delim("EffectorP_3.0.0-beta/inplantasecreted_effectorPresults.txt", sep = "\t", stringsAsFactors = FALSE, skip = 10)
effectorpinplanta.results <- effectorpinplanta.results[1:(nrow(effectorpinplanta.results) - 15), ]
colnames(effectorpinplanta.results) <- c("protein_id", "Cytoplasmic_effector", "Apoplastic_effector", "Non_effector", "Prediction")
effectorpinplanta.results$protein_id <- gsub(" ", "", effectorpinplanta.results$protein_id)

head(effectorpinplanta.results)
inplantasecreted <- effectorpinplanta.results
inplantasecreted.id <- dplyr::right_join(gtf.faa.omicsbox, inplantasecreted, by = 'protein_id')
# 2607 secreted proteins in planta

inplantanoneffector <- subset(inplantasecreted, inplantasecreted$Prediction == "Non-effector")
inplantanoneffector.id <- dplyr::right_join(gtf.faa.omicsbox, inplantanoneffector, by = 'protein_id')
# 1783 predicted non-effectors

inplantaeffectors <- subset(inplantasecreted, inplantasecreted$Prediction != "Non-effector")
inplantaeffectors.id <- dplyr::right_join(gtf.faa.omicsbox, inplantaeffectors, by = 'protein_id')
# 824 effector genes in planta
```

```{r, effector expression, echo=FALSE}

earlydetected <- as.data.frame(res_early_vs_vegetative)
earlydetected$gene_id <- rownames(earlydetected)
earlysecreted <- dplyr::right_join(earlydetected, inplantasecreted.id, by = 'gene_id')

middetected <- as.data.frame(res_middle_vs_vegetative)
middetected$gene_id <- rownames(middetected)
midsecreted <- dplyr::right_join(middetected, inplantasecreted.id, by = 'gene_id')

latedetected <- as.data.frame(res_late_vs_vegetative)
latedetected$gene_id <- rownames(latedetected)
latesecreted <- dplyr::right_join(latedetected, inplantasecreted.id, by = 'gene_id')

```

```{r, effector expression eulerr plot, echo=FALSE}

eeup <- earlysecreted[earlysecreted$log2FoldChange > 0 & earlysecreted$padj < 0.05, ]
eeup <- unique(eeup$gene_id)
eedn <- earlysecreted[earlysecreted$log2FoldChange < 0 & earlysecreted$padj < 0.05, ]
eedn <- unique(eedn$gene_id)

meup <- midsecreted[midsecreted$log2FoldChange > 0 & midsecreted$padj < 0.05, ]
meup <- unique(meup$gene_id)
medn <- midsecreted[midsecreted$log2FoldChange < 0 & midsecreted$padj < 0.05, ]
medn <- unique(medn$gene_id)

leup <- latesecreted[latesecreted$log2FoldChange > 0 & latesecreted$padj < 0.05, ]
leup <- unique(leup$gene_id)
ledn <- latesecreted[latesecreted$log2FoldChange < 0 & latesecreted$padj < 0.05, ]
ledn <- unique(ledn$gene_id)

filename <- "Effectorexpression.png"

# Create a PNG device
png(filename)

width <- 10  
height <- 6  

png(filename, width = width, height = height, units = "in", res = 300)

secreted <- list("Early_up"=eeup, "Early_down"=eedn, "Mid_up"=meup, "Mid_down"=medn, "Late_up"=leup,"Late_down"=ledn)

plot(euler(secreted),quantities = TRUE, labels = list(font = 24), main="In Planta secretome expression")

dev.off()
```

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Effectorexpression.png")
```

```{r, upregulated effector ORA, echo=FALSE}
performEnrichmentAnalysis <- function(geneSet, sets, bg) {
  ora_res <- as.data.frame(enricher(gene = geneSet,
                                    universe = bg,
                                    maxGSSize = 50000,
                                    TERM2GENE = sets,
                                    pAdjustMethod = "fdr",
                                    pvalueCutoff = 1,
                                    qvalueCutoff = 1))
  
  ora_res$geneID <- NULL
  ora_res_names <- rownames(ora_res)
  
  ora_res$GeneRatio <- as.character(ora_res$GeneRatio)
  
  gr <- as.numeric(sapply(strsplit(as.character(ora_res$GeneRatio), "/"), "[[", 1)) /
    as.numeric(sapply(strsplit(as.character(ora_res$GeneRatio), "/"), "[[", 2))
  
  br <- as.numeric(sapply(strsplit(as.character(ora_res$BgRatio), "/"), "[[", 1)) /
    as.numeric(sapply(strsplit(as.character(ora_res$BgRatio), "/"), "[[", 2))
  
  ora_res$es <- gr/br
  ora_res <- ora_res[order(-ora_res$es), ]
  ora_res$Description <- NULL
  
  return(ora_res)
}

# Define the sets and bg dataframes
bg <- detectedgenes.id$gene_id

bg.set <- data.frame("background", bg)
colnames(bg.set) <- c("term", "gene")

all.set <- data.frame("Secreted", inplantasecreted.id$gene_id)
colnames(all.set) <- c("term", "gene")

Noneffector.set <- data.frame("Non_effector", inplantanoneffector.id$gene_id)
colnames(Noneffector.set) <- c("term", "gene")

apop <- inplantasecreted.id[grep("Y", inplantasecreted.id$Apoplastic_effector),]
apop.set <- data.frame("Apoplastic", apop$gene_id)
colnames(apop.set) <- c("term", "gene")

cyto <- inplantasecreted.id[grep("Y", inplantasecreted.id$Cytoplasmic_effector),]
cyto.set <- data.frame("Cytosolic", cyto$gene_id) 
colnames(cyto.set) <- c("term", "gene")

sets <- rbind(bg.set, all.set, Noneffector.set, apop.set, cyto.set)

ESeeup <- performEnrichmentAnalysis(eeup, sets, bg)
ESmeup <- performEnrichmentAnalysis(meup, sets, bg)
ESleup <- performEnrichmentAnalysis(leup, sets, bg)

```

```{r, expression data}
dds_genes <- dds

# Extract the expression values for the top genes
dds_gene_expression <- assay(dds)[rownames(dds_genes), ]

# Create a matrix of log2-transformed expression values
log2_dds_expression <- log2(dds_gene_expression + 1)

# Convert the matrix to a dataframe
dds.df <- as.data.frame(log2_dds_expression)

# Create a new column named "gene_id" and assign the rownames to it
dds.df$gene_id <- rownames(dds.df)

# Reset the rownames of the matrix
rownames(dds.df) <- NULL

```

```{r}

rxlr <- pcrxlr$gene_id
rxlr.de <- dds.df[dds.df$gene_id %in% rxlr, ]
rownames(rxlr.de) <- rxlr.de$gene_id
rxlr.de$gene_id <- NULL

col_ann <- colData[,c(1,3)]
rownames(col_ann) <- col_ann$Sample
col_ann <- data.frame(col_ann)
col_ann$Stage <- as.factor(col_ann$Stage)
col_ann <- col_ann[order(col_ann$Stage),]
col_ann$sample_ID <- NULL

col_ann$Stage <- factor(col_ann$Stage, levels = c("Vegetative", "Early", "Middle", "Late"))

# Sort the `col_ann` dataframe based on the ordered `Stage` column
col_ann <- col_ann[order(col_ann$Stage), ]

filename <- "RxLR_normalized.png"
width <- 6400 
height <- 6400
pointsize <- 10
resolution <- 600

png(filename, width = width, height = height, pointsize = pointsize, res = resolution)

pheatmap(rxlr.de,  scale = "row",
         cluster_rows = TRUE, cluster_cols = F,
         display_numbers = F, 
         fontsize_row = 8, fontsize_col = 8, 
         annotation_col = col_ann,
         show_colnames = FALSE,
         cellheight = 8, cellwidth = 40,
         main = "RxLR- normalized counts")
dev.off()
```

```{r,fig.height=8, fig.width=12}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/RxLR_normalized.png")
```

IUM83_06578 rxlr rna helicase
IUM83_10648 rxlr nudix

```{r }
rxlr <- pcrxlr$gene_id

rxlrede <- earlydetected[earlydetected$gene_id %in% rxlr, ]
rxlrmde <- middetected[middetected$gene_id %in% rxlr, ]
rxlrlde <- latedetected[latedetected$gene_id %in% rxlr, ]

data_matrix <- matrix(
  c(rxlrede$log2FoldChange, rxlrmde$log2FoldChange, rxlrlde$log2FoldChange),
  nrow = length(rxlrede$log2FoldChange),
  ncol = 3,
  byrow = FALSE
)

rownames(data_matrix) <- rxlrede$gene_id
colnames(data_matrix) <- c("Early", "Middle", "Late")

filename <- "RxLR_Log2Fold.png"
png(filename, width = width, height = height, pointsize = pointsize, res = resolution)

# Create the heatmap with row labels
pheatmap(data_matrix,
         fontsize_row = 8, fontsize_col = 8, 
         cellheight = 8, cellwidth = 80,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         show_rownames = TRUE,
         labels_row = rxlrede$gene_id,
         main = "RxLR - log2FoldChange")
dev.off()

```

```{r}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/RxLR_Log2Fold.png")
```
IUM83_06578 rxlr rna helicase
IUM83_10648 rxlr nudix
```{r}

crn <- pccrn$gene_id
crn.de <- dds.df[dds.df$gene_id %in% crn, ]
rownames(crn.de) <- crn.de$gene_id
crn.de$gene_id <- NULL

col_ann <- colData[,c(1,3)]
rownames(col_ann) <- col_ann$Sample
col_ann <- data.frame(col_ann)
col_ann$Stage <- as.factor(col_ann$Stage)
col_ann <- col_ann[order(col_ann$Stage),]
col_ann$sample_ID <- NULL

col_ann$Stage <- factor(col_ann$Stage, levels = c("Vegetative", "Early", "Middle", "Late"))


col_ann <- col_ann[order(col_ann$Stage), ]

filename <- "CRN_normalized.png"
png(filename, width = width, height = height, pointsize = pointsize, res = resolution)

pheatmap(crn.de,  scale = "row",
         cluster_rows = TRUE, cluster_cols = F,
         display_numbers = F, 
         fontsize_row = 8, fontsize_col = 8, 
         annotation_col = col_ann,
         show_colnames = FALSE,
         cellheight = 8, cellwidth = 40,
         main = "CRN - normalized")
dev.off()
```

```{r}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/CRN_normalized.png")
```
```{r}
crnlog <- pccrn$gene_id

crnede <- earlydetected[earlydetected$gene_id %in% crnlog, ]
crnmde <- middetected[middetected$gene_id %in% crnlog, ]
crnlde <- latedetected[latedetected$gene_id %in% crnlog, ]

crn.matrix <- matrix(
  c(crnede$log2FoldChange, crnmde$log2FoldChange, crnlde$log2FoldChange),
  nrow = length(crnede$log2FoldChange),
  ncol = 3,
  byrow = FALSE
)

rownames(crn.matrix) <- crnede$gene_id
colnames(crn.matrix) <- c("Early", "Middle", "Late")

filename <- "CRN_Log2Fold.png"
png(filename, width = width, height = height, pointsize = pointsize, res = resolution)

# Create the heatmap with row labels
pheatmap(crn.matrix,
         fontsize_row = 8, fontsize_col = 8, 
         cellheight = 8, cellwidth = 80,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         show_rownames = TRUE,
         labels_row = crnede$gene_id,
         main = "CRN - log2Foldchange")
dev.off()
```

```{r}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/CRN_Log2Fold.png")
```

```{r}

nlp <- pcnlp$gene_id
nlp.de <- dds.df[dds.df$gene_id %in% nlp, ]
rownames(nlp.de) <- nlp.de$gene_id
nlp.de$gene_id <- NULL

col_ann <- colData[,c(1,3)]
rownames(col_ann) <- col_ann$Sample
col_ann <- data.frame(col_ann)
col_ann$Stage <- as.factor(col_ann$Stage)
col_ann <- col_ann[order(col_ann$Stage),]
col_ann$sample_ID <- NULL

desired_order <- c("Vegetative", "Early", "Middle", "Late")

# Order the `col_ann` dataframe based on the specified order
col_ann$Stage <- factor(col_ann$Stage, levels = desired_order)
col_ann <- col_ann[order(col_ann$Stage), ]
head(col_ann, 12)

filename <- "NLP_mormalized.png"
png(filename, width = width, height = height, pointsize = pointsize, res = resolution)

pheatmap(nlp.de,  scale = "row",
         cluster_rows = TRUE, cluster_cols = F,
         display_numbers = F, 
         fontsize_row = 8, fontsize_col = 8, 
         annotation_col = col_ann,
         show_colnames = FALSE,
         cellheight = 8, cellwidth = 40,
         main = "NLP")
dev.off()
```

```{r}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/NLP_mormalized.png")
```
```{r}
nlplog <- pcnlp$gene_id

nlpede <- earlydetected[earlydetected$gene_id %in% nlplog, ]
nlpmde <- middetected[middetected$gene_id %in% nlplog, ]
nlplde <- latedetected[latedetected$gene_id %in% nlplog, ]

nlp.matrix <- matrix(
  c(nlpede$log2FoldChange, nlpmde$log2FoldChange, nlplde$log2FoldChange),
  nrow = length(nlpede$log2FoldChange),
  ncol = 3,
  byrow = FALSE
)

rownames(nlp.matrix) <- nlpede$gene_id
colnames(nlp.matrix) <- c("Early", "Middle", "Late")

filename <- "NLP_Log2Fold.png"
png(filename, width = width, height = height, pointsize = pointsize, res = resolution)

# Create the heatmap with row labels
pheatmap(nlp.matrix,
         fontsize_row = 8, fontsize_col = 8, 
         cellheight = 8, cellwidth = 80,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         show_rownames = TRUE,
         labels_row = crnede$gene_id,
         main = "NLP - log2Foldchange")
dev.off()
```
```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/NLP_Log2Fold.png")
```
```{r}
PF00964. <-PF00964$protein_id
PF00964.id <- gtf.faa.omicsbox[gtf.faa.omicsbox$protein_id %in% PF00964., ]

pfamlog <- PF00964.id$gene_id 

pfamede <- earlydetected[earlydetected$gene_id %in% pfamlog, ]
pfammde <- middetected[middetected$gene_id %in% pfamlog, ]
pfamlde <- latedetected[latedetected$gene_id %in% pfamlog, ]

pfam.matrix <- matrix(
  c(pfamede$log2FoldChange, pfammde$log2FoldChange, pfamlde$log2FoldChange),
  nrow = length(pfamede$log2FoldChange),
  ncol = 3,
  byrow = FALSE
)

rownames(pfam.matrix) <- pfamede$gene_id
colnames(pfam.matrix) <- c("Early", "Middle", "Late")

filename <- "Pfam_Log2Fold.png"
png(filename, width = width, height = height, pointsize = pointsize, res = resolution)

# Create the heatmap with row labels
pheatmap(pfam.matrix,
         fontsize_row = 5, fontsize_col = 5, 
         cellheight = 5, cellwidth = 80,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         show_rownames = TRUE,
         labels_row = pfamede$gene_id,
         main = "Elicitin - log2Foldchange")
dev.off()

```

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Pfam_Log2Fold.png")
```

```{r, upregulated secreted lineplot, echo=FALSE}
es_apop_ESeeup <- ESeeup[ESeeup$ID == "Apoplastic", "es"]
es_cyto_ESeeup <- ESeeup[ESeeup$ID == "Cytosolic", "es"]
es_non_effector_ESeeup <- ESeeup[ESeeup$ID == "Non_effector", "es"]

es_apop_ESmeup <- ESmeup[ESmeup$ID == "Apoplastic", "es"]
es_cyto_ESmeup <- ESmeup[ESmeup$ID == "Cytosolic", "es"]
es_non_effector_ESmeup <- ESmeup[ESmeup$ID == "Non_effector", "es"]

es_apop_ESleup <- ESleup[ESleup$ID == "Apoplastic", "es"]
es_cyto_ESleup <- ESleup[ESleup$ID == "Cytosolic", "es"]
es_non_effector_ESleup <- ESleup[ESleup$ID == "Non_effector", "es"]

# Create a combined dataframe
combined_data <- data.frame(Dataset = rep(c("ESeeup", "ESmeup", "ESleup"), each = 3),
                            GeneSet = rep(c("Apoplastic", "Cytosolic", "Non_effector"), times = 3),
                            ES = c(es_apop_ESeeup, es_cyto_ESeeup, es_non_effector_ESeeup,
                                   es_apop_ESmeup, es_cyto_ESmeup, es_non_effector_ESmeup,
                                   es_apop_ESleup, es_cyto_ESleup, es_non_effector_ESleup))


dataset_order <- c("ESeeup", "ESmeup", "ESleup")
x_labels <- c("Early", "Middle", "Late")

# Convert the "Dataset" column to a factor with the desired order and labels
combined_data$Dataset <- factor(combined_data$Dataset, levels = dataset_order, labels = x_labels)

filename <- "Secreted_enrichment_scores.png"

png(filename)

width <- 8  
height <- 8  

png(filename, width = width, height = height, units = "in", res = 300)

ggplot(combined_data, aes(x = Dataset, y = ES, color = GeneSet, group = GeneSet)) +
  geom_line() +
  geom_point() +
  labs(x = "Time Point", y = "Enrichment Score (ES)",
       title = "Over-representation analysis of predicted secreted genes") +
  scale_color_manual(values = c("Cytosolic" = "blue", "Apoplastic" = "red", "Non_effector" = "black")) +
  theme_minimal()

dev.off()
```

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Secreted_enrichment_scores.png")
```

```{r}
run_enrichment <- function(gene_set, genesets, bg) {
  ora_res <- as.data.frame(enricher(gene = gene_set,
                                    universe = bg,
                                    maxGSSize = 50000,
                                    TERM2GENE = genesets,
                                    pAdjustMethod = "fdr",
                                    pvalueCutoff = 1,
                                    qvalueCutoff = 1))
  
  ora_res$geneID <- NULL
  ora_res_names <- rownames(ora_res)
  
  ora_res$GeneRatio <- as.character(ora_res$GeneRatio)
  
  gr <- as.numeric(sapply(strsplit(as.character(ora_res$GeneRatio), "/"), "[[", 1)) /
    as.numeric(sapply(strsplit(as.character(ora_res$GeneRatio), "/"), "[[", 2))
  
  br <- as.numeric(sapply(strsplit(as.character(ora_res$BgRatio), "/"), "[[", 1)) /
    as.numeric(sapply(strsplit(as.character(ora_res$BgRatio), "/"), "[[", 2))
  
  ora_res$es <- gr/br
  ora_res <- ora_res[order(-ora_res$es), ]
  ora_res$Description <- NULL
  
  top_10 <- head(ora_res[ora_res$p.adjust < 0.05, ], 10)
  
  return(top_10)
}

# Usage example
genesets <- read.gmt("goterms.gmt")
bg <- detectedgenes.id$gene_id

GOeeup <- run_enrichment(eeup, genesets, bg) 
rownames(GOeeup) <- NULL
GOeedn <- run_enrichment(eedn, genesets, bg)
rownames(GOeedn) <- NULL
GOmeup <- run_enrichment(meup, genesets, bg)
rownames(GOmeup) <- NULL
GOmedn <- run_enrichment(medn, genesets, bg)
rownames(GOmedn) <- NULL
GOleup <- run_enrichment(leup, genesets, bg)
rownames(GOleup) <- NULL
GOledn <- run_enrichment(ledn, genesets, bg)
rownames(GOledn) <- NULL
```

```{r}
#GOeeup %>% kbl(caption="GO terms Early upregulated secreted") %>%   kable_paper("hover", full_width = F)
#GOeedn %>% kbl(caption="GO terms Early downregulated secreted") %>%   kable_paper("hover", full_width = F)
#GOmeup %>% kbl(caption="GO terms Mid upregulated secreted") %>%   kable_paper("hover", full_width = F)
#GOmedn %>% kbl(caption="GO terms Mid downregulated secreted") %>%   kable_paper("hover", full_width = F)
#GOleup %>% kbl(caption="GO terms Late upregulated secreted") %>%   kable_paper("hover", full_width = F)
#GOledn %>% kbl(caption="GO terms Late downregulated secreted") %>%   kable_paper("hover", full_width = F)
```

```{r, early secreted smear}
eesig <- dplyr::right_join(earlysecreted, inplantaeffectors.id, by = 'gene_id') 
eesig <- subset(eesig, padj < 0.05)
# early significant secreted
essig <- subset(earlysecreted, padj < 0.05)
# early non-significant secreted 
esnonsig <- subset(earlysecreted, padj > 0.05)

eesup <- rownames(subset(eesig, log2FoldChange > 2 & padj < 0.05))
eesdn <- rownames(subset(eesig, log2FoldChange < 2 & padj < 0.05))
essup <- rownames(subset(essig, log2FoldChange > 2 & padj < 0.05))
essdn <- rownames(subset(essig, log2FoldChange < 2 & padj < 0.05))

# Calculate the gene counts
ESGENESUP <- length(essup)
ESGENESDN <- length(essdn)
EEGENESUP <- length(eesup)
EEGENESDN <- length(eesdn)

# Create the subtitle using the gene counts
SUBHEADER <- paste(ESGENESUP, "Secreted genes Up-regulated (", EEGENESUP, "effectors),",
                    ESGENESDN, "Secreted genes down-regulated (", EEGENESDN, "effectors)")

res_df <- data.frame(
  baseMean = res_early_vs_vegetative$baseMean,
  log2FoldChange = res_early_vs_vegetative$log2FoldChange
)

filename <- "Earlysecretedplot.png"

# Create a PNG device
png(filename)

width <- 10  
height <- 6  

png(filename, width = width, height = height, units = "in", res = 300)
ggplot(res_df, aes(x = log10(baseMean), y = log2FoldChange)) +
  geom_point(color = "gray", alpha = 0.5, size = 2, shape = 19) +
  geom_point(data = essig, aes(color = "Effector"), size = 2, shape = 19) +
  geom_point(data = eesig, aes(color = "Secreted"), size = 2, shape = 19) +
  geom_point(data = esnonsig, aes(color = "Non-significant"), size = 2, shape = 19) +
  labs(x = "log10 basemean", y = "log2 foldchange") +
  scale_color_manual(values = c("Effector" = "red", "Secreted" = "blue", "Non-significant" = "black"),
                     name = "") +
  theme_bw() +
  theme(axis.line = element_line(color = "black"),
        panel.border = element_rect(color = "black"),
        panel.grid.major = element_line(color = "grey"),  
        panel.grid.minor = element_blank(),
        legend.position = c(0.05, 0.99),  
        legend.justification = c(0, 1),  
        legend.background = element_rect(fill = NA, size = 0.5),
        legend.key.size = unit(1, "lines")) +
  ggtitle(expression(paste(italic("Phytophthora cinnamomi"), " secreted genes expressed during early interaction"))) +
  labs(subtitle = SUBHEADER) +
  theme(plot.title = element_text(face = "bold", size = 16))

dev.off()
```

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Earlysecretedplot.png")
```

```{r, middle secreted smear}
mesig <- dplyr::right_join(midsecreted, inplantaeffectors.id, by = 'gene_id') 
mesig <- subset(mesig, padj < 0.05)
# early significant secreted
mssig <- subset(midsecreted, padj < 0.05)
# early non-significant secreted 
msnonsig <- subset(midsecreted, padj > 0.05)

mesup <- rownames(subset(mesig, log2FoldChange > 2 & padj < 0.05))
mesdn <- rownames(subset(mesig, log2FoldChange < 2 & padj < 0.05))
mssup <- rownames(subset(mssig, log2FoldChange > 2 & padj < 0.05))
mssdn <- rownames(subset(mssig, log2FoldChange < 2 & padj < 0.05))

# Calculate the gene counts
mSGENESUP <- length(mssup)
mSGENESDN <- length(mssdn)
mEGENESUP <- length(mesup)
mEGENESDN <- length(mesdn)

# Create the subtitle using the gene counts
SUBHEADER <- paste(mSGENESUP, "Secreted genes Up-regulated (", mEGENESUP, "effectors),",
                    mSGENESDN, "Secreted genes down-regulated (", mEGENESDN, "effectors)")

res_df <- data.frame(
  baseMean = res_middle_vs_vegetative$baseMean,
  log2FoldChange = res_middle_vs_vegetative$log2FoldChange
)

filename <- "Midsecretedplot.png"

# Create a PNG device
png(filename)

width <- 10  
height <- 6  

png(filename, width = width, height = height, units = "in", res = 300)
ggplot(res_df, aes(x = log10(baseMean), y = log2FoldChange)) +
  geom_point(color = "gray", alpha = 0.5, size = 2, shape = 19) +
  geom_point(data = mssig, aes(color = "Effector"), size = 2, shape = 19) +
  geom_point(data = mesig, aes(color = "Secreted"), size = 2, shape = 19) +
  geom_point(data = msnonsig, aes(color = "Non-significant"), size = 2, shape = 19) +
  labs(x = "log10 basemean", y = "log2 foldchange") +
  scale_color_manual(values = c("Effector" = "red", "Secreted" = "blue", "Non-significant" = "black"),
                     name = "") +
  theme_bw() +
  theme(axis.line = element_line(color = "black"),
        panel.border = element_rect(color = "black"),
        panel.grid.major = element_line(color = "grey"),  
        panel.grid.minor = element_blank(),
        legend.position = c(0.05, 0.99),  
        legend.justification = c(0, 1),  
        legend.background = element_rect(fill = NA, size = 0.5),
        legend.key.size = unit(1, "lines")) +
  ggtitle(expression(paste(italic("Phytophthora cinnamomi"), " secreted genes expressed during middle interaction"))) +
  labs(subtitle = SUBHEADER) +
  theme(plot.title = element_text(face = "bold", size = 16))

dev.off()
```

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Midsecretedplot.png")
```

```{r, late secreted smear}
lesig <- dplyr::right_join(latesecreted, inplantaeffectors.id, by = 'gene_id') 
lesig <- subset(lesig, padj < 0.05)
# early significant secreted
lssig <- subset(latesecreted, padj < 0.05)
# early non-significant secreted 
lsnonsig <- subset(latesecreted, padj > 0.05)

lesup <- rownames(subset(lesig, log2FoldChange > 2 & padj < 0.05))
lesdn <- rownames(subset(lesig, log2FoldChange < 2 & padj < 0.05))
lssup <- rownames(subset(lssig, log2FoldChange > 2 & padj < 0.05))
lssdn <- rownames(subset(lssig, log2FoldChange < 2 & padj < 0.05))

# Calculate the gene counts
lSGENESUP <- length(lssup)
lSGENESDN <- length(lssdn)
lEGENESUP <- length(lesup)
lEGENESDN <- length(lesdn)

# Create the subtitle using the gene counts
SUBHEADER <- paste(lSGENESUP, "Secreted genes Up-regulated (", lEGENESUP, "effectors),",
                    lSGENESDN, "Secreted genes down-regulated (", lEGENESDN, "effectors)")

res_df <- data.frame(
  baseMean = res_late_vs_vegetative$baseMean,
  log2FoldChange = res_late_vs_vegetative$log2FoldChange
)

filename <- "Latesecretedplot.png"

# Create a PNG device
png(filename)

width <- 10  
height <- 6  

png(filename, width = width, height = height, units = "in", res = 300)
ggplot(res_df, aes(x = log10(baseMean), y = log2FoldChange)) +
  geom_point(color = "gray", alpha = 0.5, size = 2, shape = 19) +
  geom_point(data = mssig, aes(color = "Effector"), size = 2, shape = 19) +
  geom_point(data = mesig, aes(color = "Secreted"), size = 2, shape = 19) +
  geom_point(data = msnonsig, aes(color = "Non-significant"), size = 2, shape = 19) +
  labs(x = "log10 basemean", y = "log2 foldchange") +
  scale_color_manual(values = c("Effector" = "red", "Secreted" = "blue", "Non-significant" = "black"),
                     name = "") +
  theme_bw() +
  theme(axis.line = element_line(color = "black"),
        panel.border = element_rect(color = "black"),
        panel.grid.major = element_line(color = "grey"),  
        panel.grid.minor = element_blank(),
        legend.position = c(0.05, 0.99),  
        legend.justification = c(0, 1),  
        legend.background = element_rect(fill = NA, size = 0.5),
        legend.key.size = unit(1, "lines")) +
  ggtitle(expression(paste(italic("Phytophthora cinnamomi"), " secreted genes expressed during late interaction"))) +
  labs(subtitle = SUBHEADER) +
  theme(plot.title = element_text(face = "bold", size = 16))

dev.off()
```

```{r, echo=FALSE, out.width='100%'}
knitr::include_graphics("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Latesecretedplot.png")
```

## Session information

```{r,session}

sessionInfo()

```
