```{r, libs, message=FALSE}
library(readxl)
library("kableExtra")
library(tidyverse)
library(gridExtra)
library(reshape2)
library(dplyr)
library(edgeR)
library(clusterProfiler)
library(BiocGenerics)
library(GenomicFeatures)
library(Biostrings)
library(rtracklayer)
library(data.table)
```

download .faa file https://ftp.ncbi.nlm.nih.gov/genomes/genbank/protozoa/Phytophthora_cinnamomi/latest_assembly_versions/GCA_018691715.1_ASM1869171v1/GCA_018691715.1_ASM1869171v1_protein.faa.gz

```{r}
#setwd("~/projects/Schroeter_Pcinnamomi/PcEffector")

# Read the .faa file
faa_file <- "GCA_018691715.1_ASM1869171v1_protein.faa"  
sequences <- readAAStringSet(faa_file, format = "fasta")

formatted_sequences <- paste0(">", names(sequences), "\n", as.character(sequences), "\n")
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Pcgenes.fasta" 
writeLines(formatted_sequences, output_file)


load("Pceffetor.RData")

# first rearrange cols
allgenes <- detectedgenes
allgenes$gene_id<-rownames(allgenes)
rownames(allgenes)<-seq(nrow(allgenes))
head(allgenes)


Pcgenenames <- read.table('pcgeneids.tsv', row.names = 1, quote = "", sep='\t', fill = TRUE, header = TRUE)

pcprotein.id <- dplyr::right_join(Pcgenenames, allgenes, by = 'gene_id')


# Define a list of protein IDs to extract
protein_list <- pcprotein.id$protein_id

```

```{r}
# Read the .faa file
faa_file <- "GCA_018691715.1_ASM1869171v1_protein.faa"  
sequences <- readAAStringSet(faa_file)

# Extract headers and sequences
headers <- names(sequences)
sequence_strings <- as.character(sequences)

# Extract protein identifiers from headers
protein_identifiers <- sub("^>(\\S+).*", "\\1", headers)

# Create a data frame with protein identifiers and sequences
df <- data.frame(protein_id = protein_identifiers, Sequence = sequence_strings, stringsAsFactors = FALSE)

# Filter the "ProteinID" column to include only the first 12 characters
df$protein_id <- substr(df$protein_id, 1, 12)

# Print the updated data frame
print(df)

pcprotein.idfaa <- dplyr::right_join(df, pcprotein.id, by = 'protein_id')
pcprotein.idfaa <- pcprotein.idfaa[, 1:4]
head(pcprotein.idfaa)
```
```{r}
#.gff file for dbCAN swap column names Name and ID
pc.gff <- rtracklayer::import("https://ftp.ncbi.nlm.nih.gov/genomes/genbank/protozoa/Phytophthora_cinnamomi/latest_assembly_versions/GCA_018691715.1_ASM1869171v1/GCA_018691715.1_ASM1869171v1_genomic.gff.gz")
pc.gff.df <- as.data.frame((pc.gff))

pc.gff.df <- pc.gff.df %>%
  rename('ID' = 'Name', 'Name' = 'ID')

# Convert necessary columns to character type
pc.gff.df$ID <- as.character(pc.gff.df$Name)
pc.gff.df$Name <- as.character(pc.gff.df$ID)

# Remove specified columns
cols_to_remove <- c("collected.by", "collection.date", "country", "nat.host", "strain")
pc.gff.df <- pc.gff.df[, !(names(pc.gff.df) %in% cols_to_remove), drop = FALSE]

# Save dataframe as .gff file using fwrite
fwrite(pc.gff.df, file = "modified_pc.gff", sep = "\t", quote = FALSE)

# Save dataframe as .gff file
fwrite(pc.gff.df, file = "modified_pc.gff", sep = "\t", quote = FALSE)
```

create .fasta for inputs

```{r}
#function used to make .fasta files for analysis
write_fasta <- function(df, file_path) {
  fasta_string <- paste0(">", df$protein_id, "\n", df$Sequence, "\n")
  writeLines(fasta_string, con = file_path)
}

generate_fasta_file <- function(df, output_file) {
  write_fasta(df, output_file)
}
```

```{r}
testfasta <- head(pcprotein.idfaa)
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/test.fasta"

generate_fasta_file(testfasta, output_file)
#use for testing terminal scripts (InterProScan, SignalP and EffectorP) - 6 genes
```

```{r}
#Pc genes detect in transcriptome -- 13595 genes detected 
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/detected_genes.fasta"

generate_fasta_file(pcprotein.idfaa, output_file)

```

```{r}
#all Pc genes from .faa file -- 21335 genes 
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta"

generate_fasta_file(df, output_file)

```

###
InterProScan analysis
###
barry@ziemann-01:/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan/interproscan-5.62-94.0$ ./interproscan.sh -appl CDD,PANTHER,Pfam,PROSITEPATTERNS,PROSITEPROFILES,SFLD,SMART,TIGRFAM,Phobius,TMHMM,SUPERFAMILY -i /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta -iprlookup --goterms -b /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/interproscan_output/
05/06/2023 12:56:46:576 Welcome to InterProScan-5.62-94.0
05/06/2023 12:56:46:577 Running InterProScan v5 in STANDALONE mode... on Linux
05/06/2023 12:56:51:785 RunID: ziemann-01_20230605_125651556_b2fz
05/06/2023 12:57:02:847 Loading file /mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta
05/06/2023 12:57:02:848 Running the following analyses:
[CDD-3.20,NCBIfam-11.0,PANTHER-17.0,Pfam-35.0,Phobius-1.01,ProSitePatterns-2022_05,ProSiteProfiles-2022_05,SFLD-4,SMART-9.0,SUPERFAMILY-1.75,TMHMM-2.0c]
Available matches will be retrieved from the pre-calculated match lookup service.

Matches for any sequences that are not represented in the lookup service will be calculated locally.
05/06/2023 12:57:07:851 Uploaded 20850 unique sequences for analysis
05/06/2023 12:59:50:492 87% completed
05/06/2023 13:59:50:678 19% completed
 05/06/2023 14:59:50:824 24% completed
05/06/2023 15:59:51:007 77% completed
05/06/2023 16:12:19:464 100% done:  InterProScan analyses completed 



###
SignalP analysis
###

setwd("~/projects/Schroeter_Pcinnamomi/PcEffector/signalp-6.0/signalp6_fast/")
Sys.setenv(PATH = paste("/home/barry/.local/bin", Sys.getenv("PATH"), sep = ":"))

SIGNALP_DIR <- system("python3 -c 'import signalp; import os; print(os.path.dirname(signalp.__file__))'", intern = TRUE)

Sys.getenv("PATH")

system("signalp6 -h")

system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")


system("signalp6 --fastafile ~/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/allpcgenes.fasta --organism eukarya --output_dir ~/projects/Schroeter_Pcinnamomi/PcEffector/SignalPall_output --format none --mode fast")
Predicting: 100%|██████████| 21334/21334 [39:18<00:00,  9.05sequences/s]

```{r}
signalpresults <- read_tsv("allpcgenes.fasta.tsv")


colnames(signalpresults) <- c("protein_id", "protein_md5", "protein_length", 
                     "analysis", "signature_id", 
                     "signature_desc", "start", "end", 
                     "score", "status", "date", 
                     "interpro_id", "interpro_desc", "go_terms")
head(dipsr)
```

```{r}
library(jsonlite)

# Read the JSON file as text
json_text <- readLines("/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/allpcgenes.json", warn = FALSE)

```


```{r}
dipsr_df <- as.data.frame(dipsr)

detectedpfam <- dipsr_df %>%
  filter(analysis == "Pfam")
```



###
bdCAN analysis
###

https://bcb.unl.edu/dbCAN2/blast.php using allpcgenes.fasta

check all boxes except substrate prediction
HMMER: dbCAN (E-Value < 1e-15, coverage > 0.35)
DIAMOND: CAZy (E-Value < 1e-102
HMMER: dbCAN-sub (E-Value < 1e-15, coverage > 0.35)
CGCFinder (Distance <= 2, signature genes = CAZyme+TC)

Jinfang Zheng and others, dbCAN3: automated carbohydrate-active enzyme and substrate annotation, 
Nucleic Acids Research, 2023;, gkad328, https://doi.org/10.1093/nar/gkad328



```{r}

dbcanresults <- read_tsv("dbCAN_allpcgenes.txt")
head(dbcanresults)
```



# run on signalP_6.0 and import results 

```{r}
# Read the prediction_results.txt file into a data frame
prediction_results <- read.table("prediction_results.txt", header = F, comment.char = "#", sep = "\t")
colnames(prediction_results) <- c("protein_id", "Prediction", "OTHER", "SP(Sec/SPI)", "CS Position")
head(prediction_results)

# Filter the prediction_results data frame by "SP" in the Prediction column
signalPsecreted <- filter(prediction_results, Prediction == "SP")

# Use separate to split CS Position column
split_results <- signalPsecreted %>%
  separate(`CS Position`, into = c("cleavage_site", "probability"),
           sep = "\\. ", remove = FALSE)

split_results <- split_results %>%
  mutate(cleavage_site = gsub("CS pos: ", "", cleavage_site),
         probability = gsub("Pr: ", "", probability))

head(split_results)

# 1166/13594 predicted secreted/genes read of the reported 1366/19981 predicted secreted/predicted genes

split_results$probability <- as.numeric(split_results$probability)
hist(split_results$probability, main = "Probability Histogram", xlab = "Probability")

```

```{r}
# Read the FASTA file
fasta_data <- readDNAStringSet("signalPpredicted.fasta")

# Extract the protein ID and sequence
protein_id <- names(fasta_data)
sequence <- as.character(fasta_data)

# Create a data frame with the protein ID and sequence
fasta_df <- data.frame(protein_id = protein_id, Sequence = sequence, stringsAsFactors = FALSE)
```


```{r}
effectorp.results <- read.delim("summary_table.txt", sep = "\t", stringsAsFactors = FALSE, skip = 10)
effectorp.results <- effectorp.results[1:(nrow(effectorp.results) - 15), ]
colnames(effectorp.results) <- c("protein_id", "Cytoplasmic_effector", "Apoplastic_effector", "Non_effector", "Prediction")
effectorp.results$protein_id <- gsub(" ", "", effectorp.results$protein_id)

head(effectorp.results)

effectorp.results.id <- dplyr::right_join(pcprotein.idfaa, effectorp.results, by = 'protein_id')
head(effectorp.results.id)
```

```{r}
#save .RDS including the output from signalP and effectorP to use in PcDESeq2.rmd

#save(signalp.effector, effectorp.results.id, file = "secrete.effect.RData")

```


```{r}
gff.raw <- readGFF("detected_genes.fasta.gff3")
head(gff.raw)
view(gff.raw)
```






###RxLR

Get supplementary files from https://rdcu.be/dcczO
https://static-content.springer.com/esm/art%3A10.1186%2Fs12864-021-07552-y/MediaObjects/12864_2021_7552_MOESM2_ESM.xls

```{r, Engelbrecht supp}
pcrxlr <- read_xls('12864_2021_7552_MOESM2_ESM.xls', 
                   sheet = 'TableS2',
                   col_names = F,
                   range = 'A3:A183') 
# Rename the column
colnames(pcrxlr) <- c("gene_id")

# Remove the last three characters of each row
pcrxlr$gene_id <- substr(pcrxlr$gene_id, 1, nchar(pcrxlr$gene_id) - 3)

as.data.frame(pcrxlr)

head(pcrxlr)
```

```{r}
rxlr.df <- dplyr::right_join(effectorp.results.id, pcrxlr, by = 'gene_id')
```

run signalP and effectorP on all genes in the .faa 
```{r}
signalp.effector <- dplyr::right_join(pcprotein.idfaa, split_results, by = 'protein_id')
head(signalp.effector)

# output all genes
output_file <- "/mnt/data/barry/projects/Schroeter_Pcinnamomi/PcEffector/Raw_fasta/all_genes.fasta"

generate_fasta_file(signalp.effector, output_file)

#import results from signalp


# Read the prediction_results.txt file into a data frame
prediction_results.all <- read.table("prediction_results.all.txt", header = F, comment.char = "#", sep = "\t")
colnames(prediction_results.all) <- c("protein_id", "Prediction", "OTHER", "SP(Sec/SPI)", "CS Position")
head(prediction_results.all)

# Filter the prediction_results data frame by "SP" in the Prediction column
signalPsecreted.all <- filter(prediction_results.all, Prediction == "SP")

# Use separate to split CS Position column
split_results.all <- signalPsecreted.all %>%
  separate(`CS Position`, into = c("cleavage_site", "probability"),
           sep = "\\. ", remove = FALSE)

split_results.all <- split_results.all %>%
  mutate(cleavage_site = gsub("CS pos: ", "", cleavage_site),
         probability = gsub("Pr: ", "", probability))

head(split_results.all)

# 1166/13594 predicted secreted/genes read of the reported 1366/19981 predicted secreted/predicted genes

split_results.all$probability <- as.numeric(split_results.all$probability)
hist(split_results.all$probability, main = "Probability Histogram", xlab = "Probability")

#run throught effectorP
```



